---
title: "Cec 16S manuscript"
author: "Max Nepel"
output: 
  html_document:
    toc: true
    # toc_float: yes
comment: "analyses of 16S rRNA gene sequencing. not all analyses ended up in the manuscript. data was handeled and subset in phyloseq objects beforehand"
---

```{r, warning=FALSE, message=FALSE}
source("0_v01_common.R", chdir = TRUE) # runs the common.R file
```

# loaded libraries
```{r}
all.pkgs.loaded <- gsub("package:","", search()[grep("package:",search())])
sessionInfo()$basePkgs
pkgs.loaded <- all.pkgs.loaded[!all.pkgs.loaded %in% sessionInfo()$basePkgs]

pkgs <- data.frame(matrix(ncol = 1,nrow = 0))
  colnames(pkgs) <- c("Version")
for (i in pkgs.loaded) {
  pkgs[i,1] <- sessionInfo()$otherPkgs[[i]]$Version
  }
print(pkgs)
```

# load data sets
```{r}
physeq.Fp.abs <- readRDS(file.path(RD_DIR, "physeq.Fp.abs.rds"))
physeq.Fp.norm <- readRDS(file.path(RD_DIR, "physeq.Fp.norm.rds"))
physeq.YCp.abs <- readRDS(file.path(RD_DIR, "physeq.YCp.abs.rds"))
physeq.YCp.norm <- readRDS(file.path(RD_DIR, "physeq.YCp.norm.rds"))
physeq.FYp.abs <- readRDS(file.path(RD_DIR, "physeq.FYp.abs.rds"))
physeq.FYp.norm <- readRDS(file.path(RD_DIR, "physeq.FYp.norm.rds"))
physeq.FYp.age.abs <- readRDS(file.path(RD_DIR, "physeq.FYp.age.abs.rds"))
physeq.FYp.age.norm <- readRDS(file.path(RD_DIR, "physeq.FYp.age.norm.rds"))
physeq.Ep.mean.abs <- readRDS(file.path(RD_DIR, "physeq.Ep.mean.abs.rds"))
physeq.Ep.mean.norm <- readRDS(file.path(RD_DIR, "physeq.Ep.mean.norm.rds"))
physeq.Ep.age.abs <- readRDS(file.path(RD_DIR, "physeq.Ep.age.abs.rds"))
physeq.Ep.age.norm <- readRDS(file.path(RD_DIR, "physeq.Ep.age.norm.rds"))
physeq.p.abs <- readRDS(file.path(RD_DIR, "physeq.p.abs.rds"))
physeq.p.norm <- readRDS(file.path(RD_DIR, "physeq.p.norm.rds"))
physeq.p.alf.abs <- readRDS(file.path(RD_DIR, "physeq.p.alf.abs.rds"))
physeq.p.alf.norm <- readRDS(file.path(RD_DIR, "physeq.p.alf.norm.rds"))
physeq.p.con.abs <- readRDS(file.path(RD_DIR, "physeq.p.con.abs.rds"))
physeq.p.con.norm <- readRDS(file.path(RD_DIR, "physeq.p.con.norm.rds"))
physeq.p.xan.abs <- readRDS(file.path(RD_DIR, "physeq.p.xan.abs.rds"))
physeq.p.xan.norm <- readRDS(file.path(RD_DIR, "physeq.p.xan.norm.rds"))
```

## first glimpse on taxonomy
```{r}
(physeq4norm.ansys <- physeq.p.norm)

# no. of OTUs
nrow(tax_table(physeq4norm.ansys))

# detected phyla
length(unique(tax_table(physeq4norm.ansys)[,"Phylum"]))
levels(as.factor(tax_table(physeq4norm.ansys)[,"Phylum"]))
```

# ###########################################################################
# #################### analyses #############################################
# ###########################################################################

## 1. describe Fp
```{r}
(physeq4abs.ansys <- physeq.Fp.abs)
  (physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.Fp.norm)
  (physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.Fp" # "ds.Fp.NOxan"

table(get_variable(physeq4norm.ansys)$Ant)
sum(table(get_variable(physeq4norm.ansys)$Ant))
unique(paste0(get_variable(physeq4norm.ansys)$Tree))

```

### 1.1 alpha diversity - Fp - shannon
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

plot_richness(physeq4abs.ansys, measures = c("Shannon"))

### testing patch type alpha diversity for significance
alpha <- get_variable(physeq4abs.ansys)
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.res <- kruskal.test(Shannon ~ Ant, data = alpha))
(alpha.res.pw <- pairwise.wilcox.test(alpha$Shannon, alpha$Ant, p.adjust.method = "BH"))

p <- ggplot(alpha, aes(Ant, Shannon))
p + geom_boxplot(aes(fill = Ant), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.dots",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Fp.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)

alpha.sub <- alpha[grep("alf", alpha$Ant),]
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("con", alpha$Ant),]
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("xan", alpha$Ant),]
median(alpha.sub$Shannon)
```

### 1.2 beta diversity - Fp - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.Ant",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Fp.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```

### 1.3 drivers - Fp

#### adonis
```{r}
(m.Fp.Ant <- adonis(otu_table(physeq4norm.ansys) ~ Ant, 
                    data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))

(physeq4norm.ansys.NOxan <- subset_samples(physeq4norm.ansys, Ant != c("xan"))) # removing xanthochroa samples
  (physeq4norm.ansys.NOxan <- prune_taxa(taxa_sums(physeq4norm.ansys.NOxan) > 0, physeq4norm.ansys.NOxan))
(m.Fp.Ant.NOxan <- adonis(otu_table(physeq4norm.ansys.NOxan) ~ Ant, 
                    data = get_variable(physeq4norm.ansys.NOxan), method = "bray", permutations = 9999))
```

### 1.4 community - Fp - bar charts

#### base for stacked bar charts
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)

Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz
Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")

ab.Taxonomy.phylum <- TaxThresh(Taxonomy, tax.level = "Phylum", thresh = 0.0)
ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.5) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2)
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.2)
ab.Taxonomy.genus0.5 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.5)
ab.Taxonomy.genus0.3 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.3)
ab.Taxonomy.genus0.2 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.2)
```

#### class taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.class

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.class",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.class",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

##### get statistics - class taxonomy
```{r}
head(unique(Taxonomy$Class))
levels(factor(tax.data2plot$Taxa))

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.class.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.class.taxSample)

# get statistics min mean max
sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/Fp.min.mean.max.class.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
```

#### order taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.order

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar# + facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.order",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.order",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

##### get statistics - order taxonomy
```{r}
head(unique(Taxonomy$Order))
levels(factor(tax.data2plot$Taxa))

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.class.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.class.taxSample)

# get statistics min mean max
sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/Fp.min.mean.max.order.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
```


#### family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar # + facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.family",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.family",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)
```

#### genus taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.genus0.3

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Phylum = ab.Taxonomy$Phylum, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Family = ab.Taxonomy$Family, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# select target samples
tax.data2plot1 <- tax.data2plot0 #[-grep("unknown", tax.data2plot0$Ant), ]
```

##### all / selected tax. groups - rel.abundance per sample
```{r}
# select target tax. group

# tax.data2plot <- tax.data2plot1[grep("Bacteroidetes", tax.data2plot1$Phylum), ]
tax.data2plot <- tax.data2plot1#[grep("Alphaproteobacteria", tax.data2plot1$Class), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified.OTU_.*","_uncl", tax.data2plot$Taxa)
tax.data2plot$Taxa <- gsub("Allorhizobium−Neorhizobium−Pararhizobium−Rhizobium","Rhizobia", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Freq,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6) # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.genus",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Fp.genus",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)
```


### 1.5 core community - Fp - rel. ab. -------------------------------------- in manuscript May22
```{r}
(physeq.core <- physeq4norm.ansys)
## OTU level

### OTUs >?% in >?% of samples
{prev <- 99 # min % of samples
det <- 0.0 # min % per sample
(sub.core <- core(physeq.core, detection = det, prevalence = prev/100))
print(sub.core)}
(tax.core <- as.data.frame(tax_table(sub.core)))
unique(tax.core$Class)
tax.core$taxa <- row.names(tax.core)
# calculate average relative abundance of abundant taxa
tax.core$av.ab <- colSums(otu_table(physeq.core)[,colnames(otu_table(physeq.core)) %in% tax.core$taxa])/nrow(otu_table(physeq.core)) 
sum(tax.core$av.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample)

tax.core.sort <- tax.core[order(tax.core$Phylum, tax.core$Class, tax.core$Order, tax.core$Family, tax.core$Genus),]

write.table(tax.core.sort,
              paste(RESULTS_DIR,"/core.",plottitle,".otu.",det,".",prev,".sort.", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)

### OTUs >?% in >?% of samples
{prev <- 50 # min % of samples
det <- 0.5 # min % per sample
(sub.core <- core(physeq.core, detection = det, prevalence = prev/100))
print(sub.core)}
(tax.core <- as.data.frame(tax_table(sub.core)))
unique(tax.core$Class)
tax.core$taxa <- row.names(tax.core)
# calculate average relative abundance of abundant taxa
tax.core$av.ab <- colSums(otu_table(physeq.core)[,colnames(otu_table(physeq.core)) %in% tax.core$taxa])/nrow(otu_table(physeq.core)) 
sum(tax.core$av.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample)

tax.core.sort <- tax.core[order(tax.core$Phylum, tax.core$Class, tax.core$Order, tax.core$Family, tax.core$Genus),]

write.table(tax.core.sort,
              paste(RESULTS_DIR,"/core.",plottitle,".otu.",det,".",prev,".sort.", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)

## genus level
(physeq.genus <- aggregate_taxa(physeq.core, "Genus"))
# test <- as.data.frame(physeq.genus@tax_table@.Data)
### genera present in >95% of samples
(sub.core <- core(physeq.genus, detection = 0, prevalence = 95/100))

### genera >?% in >?% of samples
{prev <- 50 # min % of samples
det <- 1.0 # min % per sample
(sub.core <- core(physeq.genus, detection = det, prevalence = prev/100))
print(sub.core)}
(tax.core <- as.data.frame(sub.core@tax_table@.Data))
unique(tax.core$Class)

# calculate average relative abundance of abundant taxa
ab.tax.core.mat <- physeq.genus@otu_table@.Data[rownames(physeq.genus@otu_table@.Data) %in% tax.core$Genus,] # creates an OTUmatrix of only the ab.taxonomy
dim(ab.tax.core.mat)

ab.tax.core <- aggregate(rowSums(ab.tax.core.mat), by=list(tax.core$Genus), FUN=sum) # sum of al rel. abundance values grouped per ab.taxa
ab.tax.core$av.rel.ab <- ab.tax.core$x/ncol(ab.tax.core.mat) # calc average relative abundance per sample
print(ab.tax.core)
sum(ab.tax.core$av.rel.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample) 

tax.core.avAb <- merge(tax.core, ab.tax.core[,-2], by.x="Genus", by.y="Group.1")
sum(tax.core.avAb$av.rel.ab)
write.table(tax.core.avAb,
              paste(RESULTS_DIR,"/core.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)
```

### 1.6 Venn diagram - Fp - rel. ab. - as ants are not correlating significantly with beta diversity - venn() is less useful

#### genus level
```{r}
(physeq.genus <- aggregate_taxa(physeq4norm.ansys, "Genus"))
venn.var <- unique(as.character(meta(physeq.genus)$Ant))
print(venn.var)

list_core <- c() # an empty object to store information

for (n in venn.var){ # for each variable n in Ants
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq.genus, Ant == n) # Choose sample from Ants by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = 0.0, # 0.001 in atleast 90% samples 
                           prevalence = 0.95)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Ants
    list_core[[n]] <- core_m # add to a list core taxa for each group.
}

plot(venn(list_core),
     fills = c(alf="#d6e2e9", con="#cbf3f0"))

length(unique(c(list_core$alf, list_core$con, list_core$xan)))
```

#### OTU level
```{r}
venn.var <- unique(as.character(meta(physeq4norm.ansys)$Ant))
print(venn.var)

list_core <- c() # an empty object to store information

for (n in venn.var){ # for each variable n in Ants
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq4norm.ansys, Ant == n) # Choose sample from Ants by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from Ant==n 
                           detection = 0.0, # 0.001 in atleast 90% samples 
                           prevalence = 0.75)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Ants
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}

plot(venn(list_core),
     fills = c(alf="#d6e2e9", con="#cbf3f0"))

length(unique(c(list_core$alf, list_core$con, list_core$xan)))

(shared <- intersect(list_core$con, list_core$alf)) # shared between alfari and constructor
```

##### visualizing the shared OTUs as stacked bar charts
```{r}
physeqVennShared <- physeq4norm.ansys
otu_table(physeqVennShared) <- otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% shared]

OTU4tax <- as.data.frame(t(otu_table(physeqVennShared)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeqVennShared@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeqVennShared), by="Name")

ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.4)
```

###### shared OTUs as family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

## 6. Cecropia as microbe source - FY age (only trees > 1 sample) - nice idea, but problematic
```{r}
# - the idea was to test if colonized internodes per Cecropia tree have similar communities. by using adonis(Tree) we test if the dissimilarity between samples correlate with the different Cecropia trees. however, also the visible heterogeneity is expected to correlate with the Cecropia trees. also an effect by the ant species can't be rouled out ecologically. Fp of the same ant species in one Cec tree might colonized around the same time, which could have originated from the same mother colony (thinking about nuptial flight of several alates per ant colony) - to really answer that question, we would need a set of trees in which we have Fp of differnt ant species. then we could better test if the Fp communities of different ant species are similar within a tree.

(physeq4abs.ansys <- physeq.FYp.age.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.FYp.age.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.YFp.age"
```

### 6.1 beta diversity - FY age - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Tree", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))
```

## 2. describe YCp
```{r}
(physeq4abs.ansys <- physeq.YCp.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.YCp.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.Ycp.norm"

table(get_variable(physeq4norm.ansys)$Ant)
sum(table(get_variable(physeq4norm.ansys)$Ant))
length(unique(paste0(get_variable(physeq4norm.ansys)$Tree)))
```

### 2.0 alpha diversity - shannon
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

plot_richness(physeq4abs.ansys, measures = c("Shannon"))

### testing patch type alpha diversity for significance
alpha <- get_variable(physeq4abs.ansys)
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.res <- kruskal.test(Shannon ~ Ant, data = alpha))
(alpha.res.pw <- pairwise.wilcox.test(alpha$Shannon, alpha$Ant, p.adjust.method = "BH"))

p <- ggplot(alpha, aes(Ant, Shannon))
p + geom_boxplot(aes(fill = Ant), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.YCp.dots",Sys.Date(),"png",sep = "."),width = 4, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.YCp.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 4, height = 5, dpi = 300)

alpha.sub <- alpha[grep("alf", alpha$Ant),]
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("con", alpha$Ant),]
median(alpha.sub$Shannon)
```

### 2.1 beta diversity - YC - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Age
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.YC.Ant",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.YC.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```

### 2.2 drivers - YC

#### adonis
```{r}
(m.YC.Ant <- adonis(otu_table(physeq4norm.ansys) ~ Ant, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.YC.Tree <- adonis(otu_table(physeq4norm.ansys) ~ TreeR, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

#### plotting - dbRDA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ TreeR, data=get_variable(physeq4norm.ansys),  distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Age
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Tree", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))
```

### 2.3 community - YC - bar charts

#### base for stacked bar charts
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")

ab.Taxonomy.phylum <- TaxThresh(Taxonomy, tax.level = "Phylum", thresh = 0.0)
ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.5) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2)
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.2)
ab.Taxonomy.genus0.5 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.5)
ab.Taxonomy.genus0.3 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.3)
ab.Taxonomy.genus0.2 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.2)
```

#### class taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.class

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Ant"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.YC.class",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.YC.class",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

##### get statistics - class taxonomy
```{r}
head(unique(Taxonomy$Class))
levels(factor(tax.data2plot$Taxa))

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.class.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.class.taxSample)

# get statistics min mean max
sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/YCp.min.mean.max.class.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
```

#### order taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.order

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Ant"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar# + facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.YC.order",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.YC.order",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)
```

##### get statistics - order taxonomy
```{r}
head(unique(Taxonomy$Order))
levels(factor(tax.data2plot$Taxa))

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.class.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.class.taxSample)

# get statistics min mean max
sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/YCp.min.mean.max.order.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
print(d6[order(d6$mean),])
```

#### family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar # + facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.YC.family",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.YC.family",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)
```

#### genus taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.genus0.3

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Phylum = ab.Taxonomy$Phylum, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Family = ab.Taxonomy$Family, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# select target samples
tax.data2plot1 <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Ant), ]
```

##### all / selected tax. groups - rel.abundance per sample
```{r}
# select target tax. group

# tax.data2plot <- tax.data2plot1[grep("Bacteroidetes", tax.data2plot1$Phylum), ]
# tax.data2plot <- tax.data2plot1[grep("Firmicutes", tax.data2plot1$Phylum), ]
# tax.data2plot <- tax.data2plot1[grep("Actinobacteria", tax.data2plot1$Phylum), ]
# tax.data2plot <- tax.data2plot1[grep("Proteobacteria", tax.data2plot1$Phylum), ]

# tax.data2plot <- tax.data2plot1[grep("Actinobacteria", tax.data2plot1$Class), ]
tax.data2plot <- tax.data2plot1#[grep("Alphaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Betaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Flavobacteriia", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Gammaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Sphingobacteriia", tax.data2plot1$Class), ]

# tax.data2plot <- tax.data2plot1[grep("Deltaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Cyanobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Acidobacteria", tax.data2plot1$Class), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
# tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)
tax.data2plot$Taxa <- gsub("_unclassified.OTU_.*","_uncl", tax.data2plot$Taxa)
# tax.data2plot$Taxa <- gsub("Allorhizobium−Neorhizobium−Pararhizobium−Rhizobium","Rhizobia", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Freq,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6) # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.YC.genus",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.YC.genus",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)
```

### 2.4 core community - Yp - rel. ab.
```{r}
(physeq.core <- physeq4norm.ansys)
## OTU level

### OTUs >?% in >?% of samples
{prev <- 99 # min % of samples
det <- 0.0 # min % per sample
(sub.core <- core(physeq.core, detection = det, prevalence = prev/100))
print(sub.core)}
(tax.core <- as.data.frame(tax_table(sub.core)))
unique(tax.core$Class)
tax.core$taxa <- row.names(tax.core)
# calculate average relative abundance of abundant taxa
tax.core$av.ab <- colSums(otu_table(physeq.core)[,colnames(otu_table(physeq.core)) %in% tax.core$taxa])/nrow(otu_table(physeq.core)) 
sum(tax.core$av.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample)

tax.core.sort <- tax.core[order(tax.core$Phylum, tax.core$Class, tax.core$Order, tax.core$Family, tax.core$Genus),]

write.table(tax.core.sort,
              paste(RESULTS_DIR,"/core.",plottitle,".otu.",det,".",prev,".sort.", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)

{prev <- 50 # min % of samples
det <- 0.5 # min % per sample
(sub.core <- core(physeq.core, detection = det, prevalence = prev/100))
print(sub.core)}
(tax.core <- as.data.frame(tax_table(sub.core)))
unique(tax.core$Class)
tax.core$taxa <- row.names(tax.core)
# calculate average relative abundance of abundant taxa
tax.core$av.ab <- colSums(otu_table(physeq.core)[,colnames(otu_table(physeq.core)) %in% tax.core$taxa])/nrow(otu_table(physeq.core)) 
sum(tax.core$av.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample)

tax.core.sort <- tax.core[order(tax.core$Phylum, tax.core$Class, tax.core$Order, tax.core$Family, tax.core$Genus),]

write.table(tax.core.sort,
              paste(RESULTS_DIR,"/core.",plottitle,".otu.",det,".",prev,".sort.", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)
```

## 3. describe Ep
### 3.1 variation within ant colonies - age (only trees > 1 sample)
```{r}
(physeq4abs.ansys <- physeq.Ep.age.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.Ep.age.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.Ep.age"

table(get_variable(physeq4norm.ansys)$Ant)
unique(paste0(get_variable(physeq4norm.ansys)$Ant,".", get_variable(physeq4norm.ansys)$Tree))
```

#### 3.1.1 alpha diversity - Ep age - shannon
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

# plot_richness(physeq4abs.ansys, measures = c("Chao1", "Shannon", "Simpson"))
plot_richness(physeq4abs.ansys, measures = c("Shannon"))

ggsave(paste(PLOTS_DIR,"/alpha.plot.EpAge",Sys.Date(),"png",sep = "."),width = 8, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.plot.EpAge",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 6, dpi = 300)

# # plot_richness(physeq4abs.ansys, x="Age", measures = c("Shannon")) + facet_grid(~TreeR, scale = "free_x") 
# plot_richness(physeq4abs.ansys, x="Age", measures = c("Shannon")) + facet_wrap(~TreeR) + coord_cartesian(ylim = c(3.5, 5.5))
# 
# ggsave(paste(PLOTS_DIR,"/alpha.plot.EpAge.fwTree",Sys.Date(),"png",sep = "."),width = 8, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/alpha.plot.EpAge.fwTree",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE, width = 8, height = 6, dpi = 300)

plot_richness(physeq4abs.ansys, x="Age", measures = c("Shannon")) + facet_wrap(~TreeR, ncol=6)

ggsave(paste(PLOTS_DIR,"/alpha.plot.EpAge.fwTree6",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.plot.EpAge.fwTree6",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE, width = 8, height = 4, dpi = 300)

### testing patch type alpha diversity for significance
alpha <- get_variable(physeq4abs.ansys)
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.res <- kruskal.test(Shannon ~ Age, data = alpha))
(alpha.res.pw <- pairwise.wilcox.test(alpha$Shannon, alpha$Age, p.adjust.method = "BH"))

alpha.sub <- alpha[grep("\\.I\\.", alpha$Name),]
mean(alpha.sub$Shannon)
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("\\.II\\.", alpha$Name),]
mean(alpha.sub$Shannon)
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("\\.III\\.", alpha$Name),]
mean(alpha.sub$Shannon)
median(alpha.sub$Shannon)

p <- ggplot(alpha, aes(Age, Shannon))
p + geom_boxplot(aes(fill = Age), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.EpAge.dots",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.EpAge.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)

p + geom_boxplot(aes(fill = Age), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  # geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.EpAge",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.EpAge",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)
```

#### 3.1.2 beta diversity - Ep age - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Age
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Age", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.EpAge.Age",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.EpAge.Age.Ding",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)

# Age & Ant
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Age", treatment = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# ggsave(paste(PLOTS_DIR,"/plot.PCoA.EpAge.AntAge",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/plot.PCoA.EpAge.AntAge",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

#### 3.1.3 drivers - Ep age

##### adonis
```{r}
(m.EpAge.Age <- adonis(otu_table(physeq4norm.ansys) ~ Age,
                       data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.EpAge.Tree <- adonis(otu_table(physeq4norm.ansys) ~ TreeR, 
                        data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.EpAge.Ant.Age <- adonis(otu_table(physeq4norm.ansys) ~ Ant+Age,
                            data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
(m.EpAge.AntStrat.Age <- adonis(otu_table(physeq4norm.ansys) ~ Age,
                            strata = get_variable(physeq4norm.ansys)$Ant, data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

##### plotting - dbRDA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ TreeR, data=get_variable(physeq4norm.ansys),  distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

p <- PlotOrd.single.bP(bplot.df , components = c("PC1", "PC2"), groups = "Tree", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.EpAge.Tree",Sys.Date(),"png",sep = "."),width = 5.6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.EpAge.Tree",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5.6, height = 4, dpi = 300)

p <- PlotOrd.bP(bplot.df , components = c("PC1", "PC2"), groups = "Tree", treatment = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.EpAge.TreeAnt",Sys.Date(),"png",sep = "."),width = 5.6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.EpAge.TreeAnt",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5.6, height = 4, dpi = 300)
```

#### 3.1.4 community - EP age - bar charts

##### base for stacked bar charts
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")

ab.Taxonomy.phylum <- TaxThresh(Taxonomy, tax.level = "Phylum", thresh = 0.0)
ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.5) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.5)
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.5)
ab.Taxonomy.genus0.7 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.7)
ab.Taxonomy.genus0.3 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.3)
ab.Taxonomy.genus0.5 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.5)
```

##### class taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.class

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar + facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.class.fwTree",Sys.Date(),"png",sep = "."),width = 12, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.class.fwTree",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 12, height = 7, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

##### order taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.order

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar# + facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.order.fwTree",Sys.Date(),"png",sep = "."),width = 15, height = 10, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.order.fwTree",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 15, height = 10, dpi = 300)
```

##### family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.family.fwTree",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.family.fwTree",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

##### genus taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.genus0.5

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Phylum = ab.Taxonomy$Phylum, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Family = ab.Taxonomy$Family, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# select target samples
tax.data2plot1 <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Ant), ]
```

###### all / selected tax. groups - rel.abundance per sample
```{r}
# select target tax. group

# tax.data2plot <- tax.data2plot1[grep("Bacteroidetes", tax.data2plot1$Phylum), ]
# tax.data2plot <- tax.data2plot1[grep("Firmicutes", tax.data2plot1$Phylum), ]
# tax.data2plot <- tax.data2plot1[grep("Actinobacteria", tax.data2plot1$Phylum), ]
# tax.data2plot <- tax.data2plot1[grep("Proteobacteria", tax.data2plot1$Phylum), ]

# tax.data2plot <- tax.data2plot1[grep("Actinobacteria", tax.data2plot1$Class), ]
tax.data2plot <- tax.data2plot1#[grep("Alphaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Betaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Flavobacteriia", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Gammaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Sphingobacteriia", tax.data2plot1$Class), ]

# tax.data2plot <- tax.data2plot1[grep("Deltaproteobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Cyanobacteria", tax.data2plot1$Class), ]
# tax.data2plot <- tax.data2plot1[grep("Acidobacteria", tax.data2plot1$Class), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
# tax.data2plot$Taxa <- gsub("_unclassified.OTU_.*","_uncl", tax.data2plot$Taxa)
tax.data2plot$Taxa <- gsub("_unclassified.OTU_.*","_uncl", tax.data2plot$Taxa)
tax.data2plot$Taxa <- gsub("Allorhizobium−Neorhizobium−Pararhizobium−Rhizobium","Rhizobia", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Freq,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Tree, scale="free", ncol=6) # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.genus.fwTree",Sys.Date(),"png",sep = "."),width = 14, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.EpAge.genus.fwTree",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 14, height = 7, dpi = 300)
```

### 3.2 Ep averaged trees - one sample per establ. ant colony
```{r}
(physeq4abs.ansys <- physeq.Ep.mean.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.Ep.mean.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "new.ds.Ep.mean-RNAlater.only"

table(get_variable(physeq4norm.ansys)$Ant)
unique(paste0(get_variable(physeq4norm.ansys)$Ant,".", get_variable(physeq4norm.ansys)$Tree))
```

#### 3.2.1 alpha diversity - Ep.mean - shannon
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

# plot_richness(physeq4abs.ansys, measures = c("Chao1", "Shannon", "Simpson"))
plot_richness(physeq4abs.ansys, measures = c("Shannon"))

ggsave(paste(PLOTS_DIR,"/alpha.plot.Epmean",Sys.Date(),"png",sep = "."),width = 8, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.plot.Epmean",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 6, dpi = 300)

### testing patch type alpha diversity for significance
alpha <- get_variable(physeq4abs.ansys)
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.res <- wilcox.test(Shannon ~ Ant, data = alpha))

p <- ggplot(alpha, aes(Ant, Shannon))
p + geom_boxplot(aes(fill = Ant), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Epmean.dots",Sys.Date(),"png",sep = "."),width = 4, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.Epmean.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 4, height = 5, dpi = 300)

alpha.sub <- alpha[grep("alf", alpha$Ant),]
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("con", alpha$Ant),]
median(alpha.sub$Shannon)
```

#### 3.2.2 beta diversity - Ep.mean - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Ant
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.Epmean.Ant",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.Epmean.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```

#### 3.2.3 drivers - Ep.mean

##### adonis
```{r}
(m.Epmean.Ant <- adonis(otu_table(physeq4norm.ansys) ~ Ant,
                        data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

##### plotting - dbRDA
```{r}
# Ant
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ Ant, data=get_variable(physeq4norm.ansys),  distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Epmean.Ant",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.Epmean.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```

#### 3.2.4 community - Ep.mean - bar charts

##### base for stacked bar charts
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name")

ab.Taxonomy.domain <- TaxThresh(Taxonomy, tax.level = "Domain", thresh = 0.0)
ab.Taxonomy.phylum <- TaxThresh(Taxonomy, tax.level = "Phylum", thresh = 0.0)
ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.5) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.3)
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.4)
ab.Taxonomy.genus0.7 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.7)
ab.Taxonomy.genus0.3 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.3)
ab.Taxonomy.genus0.5 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.5)
```

##### domain taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.domain

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Domain, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.domain",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.domain",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)

mean.domain.Epmean <- aggregate(tax.data2plot$Rel.Ab, by=list(Taxa=tax.data2plot$Taxa), FUN=mean)
mean.domain.Epmean
```

##### phylum taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.phylum

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Phylum, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.phylum",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.phylum",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)

mean.phylum.Epmean <- aggregate(tax.data2plot$Rel.Ab, by=list(Taxa=tax.data2plot$Taxa), FUN=mean)
unique(mean.phylum.Epmean$Taxa)
mean.phylum.Epmean[order(mean.phylum.Epmean$x),]
```

##### class taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.class

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Ant"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.class",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.class",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)

# sorted by Year
AllTree.bar <- qplot(x=Tree,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllTree.bar

AllAnt.bar <- qplot(x=Ant,data=tax.data2plot,geom="bar",weight=Ant.Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllAnt.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.class.Ant",Sys.Date(),"png",sep = "."),width = 4, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.class.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)
```

###### get statistics - class taxonomy
```{r}
head(unique(Taxonomy$Class))
levels(factor(tax.data2plot$Taxa))

# Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.class.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)
head(sum.class.taxAnt) 
sum(sum.class.taxAnt$x)

# get statistics per ant species
sum.class2check <- sum.class.taxAnt # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa","alf","con", "mean") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7, 
              paste(RESULTS_DIR,"/EPmean.alf.con.mean.class.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.class.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.class.taxSample)

# get statistics min mean max
sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/EPmean.min.mean.max.class.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

# get statistics min mean max per ant species
for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot[tax.data2plot$Ant == j,]
sum.class.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
head(sum.class.taxSample)

sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, "_", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/EPmean.",j,".min.mean.max.class.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
}
```

##### order taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.order

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)
tax.data2plot$Taxa <- gsub("_or","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Ant"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Tree, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.order",Sys.Date(),"png",sep = "."),width = 10, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.order",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 10, height = 5, dpi = 300)

# sorted by Year
AllTree.bar <- qplot(x=Tree,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllTree.bar

AllAnt.bar <- qplot(x=Ant,data=tax.data2plot,geom="bar",weight=Ant.Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllAnt.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.order.Ant",Sys.Date(),"png",sep = "."),width = 6, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.order.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 5, dpi = 300)
```

###### get statistics - order taxonomy
```{r}
levels(factor(tax.data2plot$Taxa))

# Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.tax.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)

sum(sum.tax.taxAnt$x)

# get statistics per ant species
sum.tax2check <- sum.tax.taxAnt # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa","alf","con", "mean") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7, 
              paste(RESULTS_DIR,"/EPmean.alf.con.mean.order.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.tax.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.tax.taxSample)

# get statistics min mean max
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/EPmean.min.mean.max.order.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

# get statistics min mean max per ant species
for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot[tax.data2plot$Ant == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
head(sum.tax.taxSample)

sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8, 
              paste(RESULTS_DIR,"/EPmean.",j,".min.mean.max.order.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
}
```

###### try to calc which orders are significantly shifting - using wilcox.test
```{r}
df.res <- data.frame(matrix(ncol = 2,nrow = 0))
colnames(df.res) <- c("tax","p_value")
cat <- as.character(unique(ab.Taxonomy$Order))
for (i in 1:length(cat)){
target.tax <- ab.Taxonomy[grep(cat[i], ab.Taxonomy$Order),] # Rhizobiales , Verrucomicrobiales , Burkholderiales , Chitinophagales
df.res[i,1] <- cat[i]
df.res[i,2] <- wilcox.test(Freq ~ Ant, data=target.tax)$p.value
}
df.res$p_adjust <- p.adjust(df.res$p_value, method="BH")
df.res$sign_adjust <- df.res$p_adjust < 0.05
print(df.res)
```


##### family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Ant"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Ant == tax.data2plot$Ant[i]])
  tax.data2plot$Ant.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family",Sys.Date(),"png",sep = "."),width = 10, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 10, height = 6, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance

# sorted by Year
AllTree.bar <- qplot(x=Tree,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllTree.bar

AllAnt.bar <- qplot(x=Ant,data=tax.data2plot,geom="bar",weight=Ant.Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllAnt.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family.Ant",Sys.Date(),"png",sep = "."),width = 6, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 6, dpi = 300)
```

###### get statistics - family taxonomy
```{r}
levels(factor(tax.data2plot$Taxa))

# Ant (Ep) variation
tax.data2plot$taxAnt <- paste(tax.data2plot$Taxa, tax.data2plot$Ant, sep="_")
sum.tax.taxAnt <- aggregate(tax.data2plot$Ant.Rel.Ab, by=list(taxYZ=tax.data2plot$taxAnt), FUN=sum)

# get statistics per ant species
sum.tax2check <- sum.tax.taxAnt # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d7) <- c("Taxa","alf","con", "mean") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2:3] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,4] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d7)
write.table(d7, 
              paste(RESULTS_DIR,"/EPmean.alf.con.mean.family.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")
sum.tax.taxSample <- aggregate(tax.data2plot$Rel.Ab, by=list(taxYZ=tax.data2plot$taxSample), FUN=sum)
head(sum.tax.taxSample)

# get statistics min mean max
sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c("Taxa","min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6, 
              paste(RESULTS_DIR,"/EPmean.min.mean.max.family.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)

# sample variation
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep="_")

# get statistics min mean max per ant species
for (j in c("alf", "con")){
tax.data2plot.ant <- tax.data2plot[tax.data2plot$Ant == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.ant$Rel.Ab, by=list(taxYZ=tax.data2plot.ant$taxSample), FUN=sum)
head(sum.tax.taxSample)

sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, "_", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d8 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d8) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d8[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d8[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d8[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d8)
write.table(d8, 
              paste(RESULTS_DIR,"/EPmean.",j,".min.mean.max.family.", Sys.Date(),".txt", sep = ""), 
              sep="\t", row.names=FALSE)
}
```

##### genus taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.genus0.3

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Phylum = ab.Taxonomy$Phylum, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Family = ab.Taxonomy$Family, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# select target samples
tax.data2plot1 <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Ant), ]
```

###### all / selected tax. groups - rel.abundance per sample
```{r}
# select target tax. group

# tax.data2plot <- tax.data2plot1[grep("Bacteroidetes", tax.data2plot1$Phylum), ]
tax.data2plot <- tax.data2plot1#[grep("Alphaproteobacteria", tax.data2plot1$Class), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified.OTU_.*","_uncl", tax.data2plot$Taxa)
# tax.data2plot$Taxa <- gsub("Allorhizobium−Neorhizobium−Pararhizobium−Rhizobium","Rhizobia", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Freq,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Tree, scale="free", ncol=6) # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.genus0.3",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.genus0.3",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)
```

#### 3.2.5 core community - Ep.mean - rel. ab.
```{r}
(physeq.core <- physeq4norm.ansys)
## OTU level

### OTUs >?% in >?% of samples
{prev <- 90 # min % of samples
det <- 0.0 # min % per sample
(sub.core <- core(physeq.core, detection = det, prevalence = prev/100))
print(sub.core)}
mean(rowSums(otu_table(physeq4abs.ansys)))*det/100
(tax.core <- as.data.frame(sub.core@tax_table@.Data))[-c(1,2,7)]
unique(tax.core$Class)
tax.core$taxa <- row.names(tax.core)
# calculate average relative abundance of abundant taxa
tax.core$av.ab <- colSums(otu_table(physeq.core)[,colnames(otu_table(physeq.core)) %in% tax.core$taxa])/nrow(otu_table(physeq.core)) 
sum(tax.core$av.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample)

write.table(tax.core,
              paste(RESULTS_DIR,"/core.",plottitle,".otu.",det,".",prev,".", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)

## genus level
(physeq.genus <- aggregate_taxa(physeq.core, "Genus"))
### genera present in >95% of samples
(sub.core <- core(physeq.genus, detection = 0, prevalence = 95/100))

### genera >?% in >?% of samples
{prev <- 90 # min % of samples
det <- 0.0 # min % per sample
(sub.core <- core(physeq.genus, detection = det, prevalence = prev/100))
print(sub.core)}
(tax.core <- as.data.frame(sub.core@tax_table@.Data))[-c(1,2,7)]
unique(tax.core$Class)

# calculate average relative abundance of abundant taxa
ab.tax.core.mat <- physeq.genus@otu_table@.Data[rownames(physeq.genus@otu_table@.Data) %in% tax.core$Genus,] # creates an OTUmatrix of only the ab.taxonomy
dim(ab.tax.core.mat)

ab.tax.core <- aggregate(rowSums(ab.tax.core.mat), by=list(tax.core$Genus), FUN=sum) # sum of al rel. abundance values grouped per ab.taxa
ab.tax.core$av.rel.ab <- ab.tax.core$x/ncol(ab.tax.core.mat) # calc average relative abundance per sample
print(ab.tax.core)
sum(ab.tax.core$av.rel.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample) 

tax.core.avAb <- merge(tax.core, ab.tax.core[,-2], by.x="Genus", by.y="Group.1")
sum(tax.core.avAb$av.rel.ab)
write.table(tax.core.avAb,
              paste(RESULTS_DIR,"/core.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)
```

#### 3.2.6 Venn diagram - Ep.mean - rel. ab.

##### genus level
```{r}
venn.var <- unique(as.character(meta(physeq.genus)$Ant))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 75 # min % of samples
det <- 0.5 # min % per sample

for (n in venn.var){ # for each variable n in Ants
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq.genus, Ant == n) # Choose sample from Ants by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Ants
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}

plot(venn(list_core),
     fills = c(alf="#d6e2e9", con="#cbf3f0"))

length(intersect(list_core$con, list_core$alf)) # no. of shared 
length(unique(c(list_core$alf, list_core$con))) # no. of all valid taxa
length(intersect(list_core$con, list_core$alf))/length(unique(c(list_core$alf, list_core$con)))*100

(shared <- intersect(list_core$con, list_core$alf)) # shared
(alf <- setdiff(list_core$alf, list_core$con)) # only in alf
(con <- setdiff(list_core$con, list_core$alf)) # only in con

sq <- seq(max(length(shared), length(alf), length(con)))
d9 <- data.frame(shared[sq], alf[sq], con[sq])
colnames(d9) <- c("shared","alf.only","con.only") # per site
print(d9)
write.table(d9,
              paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)
```

##### OTU level - widespread OTUs
```{r}
venn.var <- unique(as.character(meta(physeq4norm.ansys)$Ant))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 90 # min % of samples
det <- 0.0 # min % per sample

for (n in venn.var){ # for each variable n in Ants
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq4norm.ansys, Ant == n) # Choose sample from Ants by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Ants
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}

# subset the EPmean dataset extracting only widespread OTUs
(abundant.otus <- unique(c(list_core[[1]], list_core[[2]])))
length(abundant.otus) # No. of OTUs widespread in alf or con EPs
df.abundant <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% abundant.otus,]
head(df.abundant)

# average relative abundance of every OTU in the EPmean dataset
df.abundant$av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% abundant.otus])/nrow(otu_table(physeq4norm.ansys))
head(df.abundant)
(abundant.percOfTotReads <- sum(df.abundant$av.ab)) # % of EPmean reads

plot(venn(list_core),
     fills = c(alf="#d6e2e9", con="#cbf3f0"))

length(intersect(list_core$con, list_core$alf)) # no. of shared 
length(unique(c(list_core$alf, list_core$con))) # no. of all valid taxa
length(intersect(list_core$con, list_core$alf))/length(unique(c(list_core$alf, list_core$con)))*100

(shared <- intersect(list_core$con, list_core$alf)) # shared

# shared OTUs account for X% of EPmean reads
shared.av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% shared])/
  nrow(otu_table(physeq4norm.ansys))
sum(shared.av.ab)

(alf <- setdiff(list_core$alf, list_core$con)) # only in alf
length(alf)
(con <- setdiff(list_core$con, list_core$alf)) # only in con
length(con)


(df.shared <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% shared,])
df.shared$av.ab <- shared.av.ab
df.shared.sort <- df.shared[order(df.shared$Phylum, df.shared$Class, df.shared$Order, df.shared$Family, df.shared$Genus),]
unique(df.shared.sort$Class)
write.table(df.shared.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".shared.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=TRUE)
(df.alf <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% alf,])
df.alf$av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% alf])/
  nrow(otu_table(physeq4norm.ansys))
df.alf.sort <- df.alf[order(df.alf$Phylum, df.alf$Class, df.alf$Order, df.alf$Family, df.alf$Genus),]
unique(df.alf.sort$Class)
write.table(df.alf.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".alf.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=TRUE)
(df.con <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% con,])
df.con$av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% con])/
  nrow(otu_table(physeq4norm.ansys))
df.con.sort <- df.con[order(df.con$Phylum, df.con$Class, df.con$Order, df.con$Family, df.con$Genus),]
unique(df.con.sort$Class)
write.table(df.con.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".con.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=TRUE)
df.widespread.sort <- cbind(
    "OTU" = c(rownames(df.shared.sort), rownames(df.alf.sort), rownames(df.con.sort)),
    "category" =  c(rep("Shared",length(shared)), rep("A. alfari",length(alf)), rep("A. constructor",length(con))),
    rbind(df.shared.sort, df.alf.sort, df.con.sort)
    )
write.table(df.widespread.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".all3Sorted.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

##### OTU level - prevalent OTUs
```{r}
venn.var <- unique(as.character(meta(physeq4norm.ansys)$Ant))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 50 # min % of samples
det <- 0.5 # min % per sample

for (n in venn.var){ # for each variable n in Ants
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq4norm.ansys, Ant == n) # Choose sample from Ants by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Ants
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}

# subset the EPmean dataset extracting only widespread OTUs
(abundant.otus <- unique(c(list_core[[1]], list_core[[2]])))
length(abundant.otus) # No. of OTUs widespread in alf or con EPs
df.abundant <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% abundant.otus,]
head(df.abundant)

# average relative abundance of every OTU in the EPmean dataset
df.abundant$av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% abundant.otus])/nrow(otu_table(physeq4norm.ansys))
head(df.abundant)
(abundant.percOfTotReads <- sum(df.abundant$av.ab)) # % of EPmean reads

plot(venn(list_core),
     fills = c(alf="#d6e2e9", con="#cbf3f0"))

length(intersect(list_core$con, list_core$alf)) # no. of shared 
length(unique(c(list_core$alf, list_core$con))) # no. of all valid taxa
length(intersect(list_core$con, list_core$alf))/length(unique(c(list_core$alf, list_core$con)))*100

(shared <- intersect(list_core$con, list_core$alf)) # shared

# shared OTUs account for X% of EPmean reads
shared.av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% shared])/
  nrow(otu_table(physeq4norm.ansys))
sum(shared.av.ab)

(alf <- setdiff(list_core$alf, list_core$con)) # only in alf
length(alf)
(con <- setdiff(list_core$con, list_core$alf)) # only in con
length(con)

(df.shared <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% shared,])
df.shared$av.ab <- shared.av.ab
df.shared.sort <- df.shared[order(df.shared$Phylum, df.shared$Class, df.shared$Order, df.shared$Family, df.shared$Genus),]
unique(df.shared.sort$Class)
write.table(df.shared.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".shared.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=TRUE)
(df.alf <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% alf,])
df.alf$av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% alf])/
  nrow(otu_table(physeq4norm.ansys))
df.alf.sort <- df.alf[order(df.alf$Phylum, df.alf$Class, df.alf$Order, df.alf$Family, df.alf$Genus),]
unique(df.alf.sort$Class)
write.table(df.alf.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".alf.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=TRUE)
(df.con <- as.data.frame(physeq4norm.ansys@tax_table@.Data)[rownames(physeq4norm.ansys@tax_table) %in% con,])
df.con$av.ab <- colSums(otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% con])/
  nrow(otu_table(physeq4norm.ansys))
df.con.sort <- df.con[order(df.con$Phylum, df.con$Class, df.con$Order, df.con$Family, df.con$Genus),]
unique(df.con.sort$Class)
write.table(df.con.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".con.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=TRUE)
df.widespread.sort <- cbind(
    "OTU" = c(rownames(df.shared.sort), rownames(df.alf.sort), rownames(df.con.sort)),
    "category" =  c(rep("Shared",length(shared)), rep("A. alfari",length(alf)), rep("A. constructor",length(con))),
    rbind(df.shared.sort, df.alf.sort, df.con.sort)
    )
write.table(df.widespread.sort,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".all3Sorted.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

###### visualizing the shared OTUs as stacked bar charts
```{r}
physeqVennShared <- physeq4norm.ansys
otu_table(physeqVennShared) <- otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% shared]

OTU4tax <- as.data.frame(t(otu_table(physeqVennShared)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeqVennShared@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeqVennShared), by="Name")

ab.Taxonomy.genus <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.4)
```

####### shared OTUs as genus taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.genus

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.genus",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.genus",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

## 4. succession - patch (Ep tree means) -------------- type: sign. R2: 0.20 -----
```{r}
(physeq4abs.ansys <- physeq.p.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.p.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.p.newEpmean"

# adding a new category for splitting up EP for analysing IP, YP, EP_alf, EP_con
sample_data(physeq4norm.ansys)$Succ <- paste0(sample_data(physeq4norm.ansys)$Type, "_", sample_data(physeq4norm.ansys)$Ant)
sample_data(physeq4norm.ansys)$Succ <- gsub("0\\.Fp_.*","0\\.Fp", sample_data(physeq4norm.ansys)$Succ)
sample_data(physeq4norm.ansys)$Succ <- as.factor(gsub("1\\.YCp_.*","1\\.YCp", sample_data(physeq4norm.ansys)$Succ))

table(get_variable(physeq4norm.ansys)$Ant)
sum(table(get_variable(physeq4norm.ansys)$Ant))
length(unique(paste0(get_variable(physeq4norm.ansys)$Tree)))
```

### 4.1 alpha diversity - shannon - all colonies
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

plot_richness(physeq4abs.ansys, measures = c("Shannon"))

ggsave(paste(PLOTS_DIR,"/alpha.plot.p",Sys.Date(),"png",sep = "."),width = 10, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.plot.p",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 10, height = 6, dpi = 300)

### testing patch type alpha diversity for significance
alpha <- get_variable(physeq4abs.ansys)
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.res <- kruskal.test(Shannon ~ Type, data = alpha))
(alpha.res.pw <- pairwise.wilcox.test(alpha$Shannon, alpha$Type, p.adjust.method = "BH"))

alpha.sub <- alpha[grep("0.Fp", alpha$Type),]
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("1.YCp", alpha$Type),]
median(alpha.sub$Shannon)
alpha.sub <- alpha[grep("2.Ep", alpha$Type),]
median(alpha.sub$Shannon)

p <- ggplot(alpha, aes(Type, Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.dots",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)

p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  # geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)
```

### 4.2 alpha diversity - shannon - intra ant species
```{r}
### testing patch type alpha diversity for significance
alpha <- get_variable(physeq4abs.ansys)
alpha$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon
alpha$AntType <- paste0(alpha$Ant, "_", alpha$Type)
(alpha.intra.res <- kruskal.test(Shannon ~ AntType, data = alpha))

### testing for pairwise significance
(alpha.intra.res.pw <- pairwise.wilcox.test(alpha$Shannon, alpha$AntType, p.adjust.method = "BH"))

### getting significance letters
(a <- melt(alpha.intra.res.pw$p.value))
a.cc  <-  na.omit(a)
a.pvals  <-  a.cc[, 3]
names(a.pvals)  <-  paste(a.cc[, 2], a.cc[, 1], sep="-")
a.pvals
(sign.letters <- multcompView::multcompLetters(a.pvals))

### plotting alpa diversities - Shannon
p <- ggplot(alpha, aes(AntType, Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.AntType",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.AntType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)

### summary
cat <- sort(unique(alpha$AntType))
df.all <- data.frame(NULL)
for (k in 1:length(cat)){
alpha.sub <- alpha[grep(cat[k], alpha$AntType),]
df.all[k,1] <- cat[k]
df.all[k,2] <- median(alpha.sub$Shannon)
}
colnames(df.all) <- c("AntType","median.Shannon")
df.all$SignLetter <- sign.letters$Letters
print(df.all)

write.table(df.all,
            paste(RESULTS_DIR,"/alpha.boxplot.p.summary.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

### 4.2 beta diversity - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.Type",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.Type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)

# Ant
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Ant", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.Ant",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
# ggCsave(paste(PLOTS_DIR,"/plot.PCoA.p.Ant",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)

# Type & Ant
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.AntType",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.AntType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)

# Type & Type
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.TypeType",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.TypeType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

### 4.3 drivers

#### adonis
```{r}
(m.p.Type <- adonis(otu_table(physeq4norm.ansys) ~ Type,
                    data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

#### plotting - dbRDA
```{r}
dbRDA.obj <- capscale((otu_table(physeq4norm.ansys)) ~ Type, data=get_variable(physeq4norm.ansys),  distance = "bray")

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.p.Type",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.p.Type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```

### 4.4 community - bar charts

#### base for stacked bar charts
```{r}
OTU4tax <- as.data.frame(t(otu_table(physeq4norm.ansys)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeq4norm.ansys@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeq4norm.ansys), by="Name") #meta

ab.Taxonomy.domain <- TaxThresh(Taxonomy, tax.level = "Domain", thresh = 0.0)
ab.Taxonomy.phylum <- TaxThresh(Taxonomy, tax.level = "Phylum", thresh = 0.0)
ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.27)
ab.Taxonomy.order.3 <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.3)
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.2)
ab.Taxonomy.genus0.5 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.5)
ab.Taxonomy.genus0.3 <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.3)
```

##### domain taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.domain

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Domain, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Ant, scale="free", ncol=6)

ggsave(paste(PLOTS_DIR,"/bar.tax.p.domain",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.domain",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 5, dpi = 300)

mean.domain.Epmean <- aggregate(tax.data2plot$Rel.Ab, by=list(Taxa=tax.data2plot$Taxa), FUN=mean)
mean.domain.Epmean
```

#### plot class taxonomy - samples
```{r}
ab.Taxonomy <- ab.Taxonomy.class

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, 
                             Freq = ab.Taxonomy$Freq, 
                             Taxa = ab.Taxonomy$Class, 
                             Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, 
                             Succ = ab.Taxonomy$Succ)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Succ"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Type == tax.data2plot$Type[i]])
  tax.data2plot$Type.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "succession.Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Succ == tax.data2plot$Succ[i]])
  tax.data2plot$Succ.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.class",Sys.Date(),"png",sep = "."),width = 14, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.class",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 14, height = 5, dpi = 300)

AllSample.bar + facet_wrap( ~ Type, scale="free_x", ncol=3) # rel. Abundance
```

##### plot class taxonomy - per Type
```{r}
  ## plot stacked bar chart per Type
Type.bar <- qplot(x=Type,data=tax.data2plot,geom="bar",weight=Type.Rel.Ab,fill=Taxa,main=plottitle) +
  scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Type.bar + facet_wrap( ~ Type, scale="free_x") # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.type",Sys.Date(),"png",sep = "."),width = 6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 4, dpi = 300)
```

###### per Type - get statistics
```{r}
head(unique(Taxonomy$Class)[order(unique(Taxonomy$Class))])
levels(factor(tax.data2plot$Taxa))

# Type (p) variation
tax.data2plot$taxType <- paste(tax.data2plot$Taxa, tax.data2plot$Type, sep=";")
sum.class.taxType <- aggregate(tax.data2plot$Type.Rel.Ab, by=list(taxYZ=tax.data2plot$taxType), FUN=sum)
sum.class.taxType 
sum(sum.class.taxType$x)

# get statistics per ant species
sum.class2check <- sum.class.taxType # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, ";", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 5,nrow = 0))
  colnames(d7) <- c("Taxa", "mean","IP","YC", "EP")
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d7[k,2] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d7[k,3:5] <- sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]]
}
  print(d7)
write.table(d7,
              paste(RESULTS_DIR,"/p.types.mean.class.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# sample variation within patch types
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep=";")

# get statistics min mean max per type species
for (j in c("0.Fp", "1.YCp", "2.Ep")){
tax.data2plot.type <- tax.data2plot[tax.data2plot$Type == j,]
sum.class.taxSample <- aggregate(tax.data2plot.type$Rel.Ab, by=list(taxYZ=tax.data2plot.type$taxSample), FUN=sum)
head(sum.class.taxSample)

sum.class2check <- sum.class.taxSample # sum.class.taxType
split <- str_split_fixed(sum.class2check$taxYZ, ";", 2)
sum.class.min.max.2 <- cbind(split,sum.class2check)
colnames(sum.class.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.class.min.max.2["Taxa"] <- lapply(sum.class.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.class.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.class.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.class.min.max.2$rel.ab[sum.class.min.max.2$Taxa == levels(sum.class.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6,
              paste(RESULTS_DIR,"/p.",j,".min.mean.max.class.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
```

##### plot class taxonomy - per SuccessionType
```{r}
# split up between EP ants
  ## plot stacked bar chart per Succession.Type
Succ.bar <- qplot(x=Succ,data=tax.data2plot,geom="bar",weight=Succ.Rel.Ab,fill=Taxa,main=plottitle) +
  scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Succ.bar + facet_wrap( ~ Type, scale="free_x") # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.Succ.fwType",Sys.Date(),"png",sep = "."),width = 6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.Succ.fwType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 4, dpi = 300)

Succ.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.SuccType",Sys.Date(),"png",sep = "."),width = 6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.SuccType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 4, dpi = 300)

# mOm mean of 2 means (EP_alf, EP_con) = merged EP after splitting up between EP ants

EPOnly <- tax.data2plot[grep("2.Ep",tax.data2plot$Type),]
EPOnly$Succ.Rel.Ab <- EPOnly$Succ.Rel.Ab/2
NoEP <- tax.data2plot[-grep("2.Ep",tax.data2plot$Type),]
tax.data2plotmOm <- rbind(NoEP,EPOnly)

Succ.bar <- qplot(x=Type,data=tax.data2plotmOm,geom="bar",weight=Succ.Rel.Ab,fill=Taxa,main=paste0(plottitle,".mOm.SuccType")) +
  scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Succ.bar + facet_wrap( ~ Type, scale="free_x") # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.mOm.Succ.fwType",Sys.Date(),"png",sep = "."),width = 6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.mOm.Succ.fwType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 4, dpi = 300)

Succ.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.mOm.SuccType",Sys.Date(),"png",sep = "."),width = 6, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.class.mOm.SuccType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 6, height = 4, dpi = 300)
```

###### per SuccessionType - get statistics
```{r}
head(unique(Taxonomy$Class))
levels(factor(tax.data2plot$Taxa))

# Succ (p) variation
tax.data2plot$taxSucc <- paste(tax.data2plot$Taxa, tax.data2plot$Succ, sep=";")
sum.tax.taxSucc <- aggregate(tax.data2plot$Succ.Rel.Ab, by=list(taxYZ=tax.data2plot$taxSucc), FUN=sum)

# get statistics per SuccessionType
sum.tax2check <- sum.tax.taxSucc
split <- str_split_fixed(sum.tax2check$taxYZ, ";", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 6,nrow = 0))
  colnames(d7) <- c("Taxa", "IP","YC", "EP_alf", "EP_con", "mOm_EP")
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  # d7[k,2] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d7[k,2:5] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
  d7[k,6] <- (d7[k,4] + d7[k,5])/2 # calc mOm = mean of means
}
  print(d7)
write.table(d7,
              paste(RESULTS_DIR,"/p.Succ.mean.class.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# sample variation within Successiontypes
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep=";")

# get statistics min mean max per Succtype species
for (j in c("0.Fp", "1.YCp", "2.Ep_alf", "2.Ep_con")){
tax.data2plot.type <- tax.data2plot[tax.data2plot$Succ == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.type$Rel.Ab, by=list(taxYZ=tax.data2plot.type$taxSample), FUN=sum)
head(sum.tax.taxSample)

sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, ";", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6,
              paste(RESULTS_DIR,"/p.",j,".min.mean.max.class.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
```

#### plot order taxonomy - samples
```{r}
ab.Taxonomy <- ab.Taxonomy.order.3

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, 
                             Freq = ab.Taxonomy$Freq, 
                             Taxa = ab.Taxonomy$Order, 
                             Class=ab.Taxonomy$Class, 
                             Type = ab.Taxonomy$Type, 
                             Ant = ab.Taxonomy$Ant, 
                             Succ = ab.Taxonomy$Succ)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Succ"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Type == tax.data2plot$Type[i]])
  tax.data2plot$Type.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "succession.Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Succ == tax.data2plot$Succ[i]])
  tax.data2plot$Succ.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.3",Sys.Date(),"png",sep = "."),width = 14, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.3",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 14, height = 5, dpi = 300)

AllSample.bar + facet_wrap( ~ Type, scale="free_x", ncol=3)

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.3.fwType",Sys.Date(),"png",sep = "."),width = 14, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.3.fwType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 14, height = 5, dpi = 300)
```

##### plot order taxonomy - per Type
```{r}
## plot stacked bar chart per Type
Type.bar <- qplot(x=Type,data=tax.data2plot,geom="bar",weight=Type.Rel.Ab,fill=Taxa,main=plottitle) +
  scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Type.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.3.type",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.3.type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 4, dpi = 300)
```

###### per Type - get statistics
```{r}
# Type (p) variation
tax.data2plot$taxType <- paste(tax.data2plot$Taxa, tax.data2plot$Type, sep=";")
sum.tax.taxType <- aggregate(tax.data2plot$Type.Rel.Ab, by=list(taxYZ=tax.data2plot$taxType), FUN=sum)

# get statistics per Type
sum.tax2check <- sum.tax.taxType # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, ";", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 5,nrow = 0))
  colnames(d7) <- c("Taxa", "mean","IP","YC", "EP")
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d7[k,2] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d7[k,3:5] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
}
  print(d7)
write.table(d7,
              paste(RESULTS_DIR,"/p.types.mean.order.3.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# sample variation within patch types
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep=";")

# get statistics min mean max per type species
for (j in c("0.Fp", "1.YCp", "2.Ep")){
tax.data2plot.type <- tax.data2plot[tax.data2plot$Type == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.type$Rel.Ab, by=list(taxYZ=tax.data2plot.type$taxSample), FUN=sum)
head(sum.tax.taxSample)

sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, ";", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6,
              paste(RESULTS_DIR,"/p.",j,".min.mean.max.order.3.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
```

##### plot order taxonomy - per SuccessionType
```{r}
# split up between EP ants
  ## plot stacked bar chart per Succession.Type
Succ.bar <- qplot(x=Succ,data=tax.data2plot,geom="bar",weight=Succ.Rel.Ab,fill=Taxa,main=plottitle) +
  scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Succ.bar + facet_wrap( ~ Type, scale="free_x") # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.fw.SuccType",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.fw.SuccType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 4, dpi = 300)

Succ.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.SuccType",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.SuccType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 4, dpi = 300)

# mOm mean of 2 means (EP_alf, EP_con) = merged EP after splitting up between EP ants

EPOnly <- tax.data2plot[grep("2.Ep",tax.data2plot$Type),]
EPOnly$Succ.Rel.Ab <- EPOnly$Succ.Rel.Ab/2
NoEP <- tax.data2plot[-grep("2.Ep",tax.data2plot$Type),]
tax.data2plotmOm <- rbind(NoEP,EPOnly)

Succ.bar <- qplot(x=Type,data=tax.data2plotmOm,geom="bar",weight=Succ.Rel.Ab,fill=Taxa,main=paste0(plottitle,".mOm.SuccType")) +
  scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Succ.bar + facet_wrap( ~ Type, scale="free_x") # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.fw.mOm.SuccType",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.fw.mOm.SuccType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 4, dpi = 300)

Succ.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.mOm.SuccType",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.order.mOm.SuccType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 4, dpi = 300)
```

###### per SuccessionType - get statistics
```{r}
head(unique(Taxonomy$Order))
levels(factor(tax.data2plot$Taxa))

# Succ (p) variation
tax.data2plot$taxSucc <- paste(tax.data2plot$Taxa, tax.data2plot$Succ, sep=";")
sum.tax.taxSucc <- aggregate(tax.data2plot$Succ.Rel.Ab, by=list(taxYZ=tax.data2plot$taxSucc), FUN=sum)

# get statistics per SuccessionType
sum.tax2check <- sum.tax.taxSucc
split <- str_split_fixed(sum.tax2check$taxYZ, ";", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d7 <- data.frame(matrix(ncol = 6,nrow = 0))
  colnames(d7) <- c("Taxa", "IP","YC", "EP_alf", "EP_con", "mOm_EP")
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d7[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  # d7[k,2] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d7[k,2:5] <- sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]]
    d7[k,6] <- (d7[k,4] + d7[k,5])/2 # calc mOm = mean of means
}
  print(d7)
write.table(d7,
              paste(RESULTS_DIR,"/p.Succ.mean.order.3.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)

# sample variation within Successiontypes
tax.data2plot$taxSample <- paste(tax.data2plot$Taxa, tax.data2plot$Sample, sep=";")

# get statistics min mean max per Succtype species
for (j in c("0.Fp", "1.YCp", "2.Ep_alf", "2.Ep_con")){
tax.data2plot.type <- tax.data2plot[tax.data2plot$Succ == j,]
sum.tax.taxSample <- aggregate(tax.data2plot.type$Rel.Ab, by=list(taxYZ=tax.data2plot.type$taxSample), FUN=sum)
head(sum.tax.taxSample)

sum.tax2check <- sum.tax.taxSample # sum.tax.taxType
split <- str_split_fixed(sum.tax2check$taxYZ, ";", 2)
sum.tax.min.max.2 <- cbind(split,sum.tax2check)
colnames(sum.tax.min.max.2) <- c("Taxa", "YZ", "taxYZ", "rel.ab")
sum.tax.min.max.2["Taxa"] <- lapply(sum.tax.min.max.2["Taxa"], as.factor)

d6 <- data.frame(matrix(ncol = 4,nrow = 0))
  colnames(d6) <- c(paste0(j,".Taxa"),"min","mean", "max") # per site
for (k in 1:length(levels(sum.tax.min.max.2$Taxa))) {
  d6[k,1] <- levels(sum.tax.min.max.2$Taxa)[[k]]
  d6[k,2] <- min(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,3] <- mean(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
  d6[k,4] <- max(sum.tax.min.max.2$rel.ab[sum.tax.min.max.2$Taxa == levels(sum.tax.min.max.2$Taxa)[[k]]])
}
  print(d6)
write.table(d6,
              paste(RESULTS_DIR,"/p.",j,".min.mean.max.order.3.", Sys.Date(),".txt", sep = ""),
              sep="\t", row.names=FALSE)
}
```

#### plot family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Extraction = ab.Taxonomy$Extraction)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Extraction"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl.", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## fuegt relative abundance fuer OTU innerhalb "Type"
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Type == tax.data2plot$Type[i]])
  tax.data2plot$Type.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar # + facet_wrap( ~ Type, scale="free", ncol=2)

ggsave(paste(PLOTS_DIR,"/bar.tax.p.family",Sys.Date(),"png",sep = "."),width = 14, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.family",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 14, height = 7, dpi = 300)

  ## plot stacked bar chart per Type
Type.bar <- qplot(x=Type,data=tax.data2plot,geom="bar",weight=Type.Rel.Ab,fill=Taxa,main=plottitle) +
  scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
Type.bar #+ facet_wrap( ~ Type, scale="free") # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.family.type",Sys.Date(),"png",sep = "."),width = 8, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.family.type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 8, height = 4, dpi = 300)
```

#### plot genus taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.genus0.3

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Genus, Phylum = ab.Taxonomy$Phylum, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Family = ab.Taxonomy$Family, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant)

# select target samples
tax.data2plot1 <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Ant), ]
```

##### all / selected tax. groups - rel.abundance per sample
```{r}
# select target tax. group

# tax.data2plot <- tax.data2plot1[grep("Bacteroidetes", tax.data2plot1$Phylum), ]
tax.data2plot <- tax.data2plot1#[grep("Alphaproteobacteria", tax.data2plot1$Class), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified.OTU_.*","_uncl", tax.data2plot$Taxa)
tax.data2plot$Taxa <- gsub("Allorhizobium−Neorhizobium−Pararhizobium−Rhizobium","Rhizobia", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Freq,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar #+ facet_wrap( ~ Type, scale="free", ncol=2) # rel. Abundance

ggsave(paste(PLOTS_DIR,"/bar.tax.p.genus",Sys.Date(),"png",sep = "."),width = 14, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.p.genus",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 14, height = 7, dpi = 300)
```

### 4.5 ternary plot
#### 4.5.1 all OTUs
```{r}
# physeq4norm.ansys_merged <- merge_samples(physeq4norm.ansys, "Type", fun = mean) # merge by source
# physeq4norm.ansys_merged_rel <- transform_sample_counts(physeq4norm.ansys_merged, function(x) x / sum(x)*100 ) # rel abundance per type
# 
# meandf <- as(otu_table(physeq4norm.ansys_merged_rel), "matrix")
# if (!taxa_are_rows(physeq4norm.ansys_merged_rel)) { meandf <- t(meandf) }
# abundance <- rowSums(meandf) / sum(meandf) * 100
# p4Ternary <- data.frame(
#   meandf,
#   Abundance = abundance,
#   Phylum = tax_table(physeq4norm.ansys_merged_rel)[, "Phylum"]
# )
# 
# p_ternary_p <-
#   ggtern(data = p4Ternary,
#          aes(
#            x = X0.Fp,
#            y = X1.YCp,
#            z = X2.Ep,
#            size = Abundance,
#            colour = Phylum
#          )) +
#   geom_point(alpha = 1 / 2) +
#   scale_size(
#     range = c(1, 10),
#     name = "Abundance (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(plottitle," - OTUs")) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# 
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".otu.", Sys.Date(),".png", sep=""),
#        width = 7, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".otu.", Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

#### 4.5.2 all genera
```{r}
# (physeq4norm.ansys.genus <- aggregate_taxa(physeq4norm.ansys, "Genus"))
# (physeq4norm.ansys.genus_merged <- merge_samples(physeq4norm.ansys.genus, "Type", fun = mean)) # merge by type
# (physeq4norm.ansys.genus_merged_rel <- transform_sample_counts(physeq4norm.ansys.genus_merged, function(x) x / sum(x) * 100 )) # rel abundance per type
# 
# meandf <- as(otu_table(physeq4norm.ansys.genus_merged_rel), "matrix")
# if (!taxa_are_rows(physeq4norm.ansys.genus_merged_rel)) { meandf <- t(meandf) }
# abundance <- rowSums(meandf) / sum(meandf) * 100
# p4Ternary <- data.frame(
#   meandf,
#   Abundance = abundance,
#   Phylum = tax_table(physeq4norm.ansys.genus_merged_rel)[, "Phylum"]
#   )
# 
# p_ternary_p <-
#   ggtern(data = p4Ternary,
#          aes(
#            x = X0.Fp,
#            y = X1.YCp,
#            z = X2.Ep,
#            size = Abundance,
#            colour = Phylum
#          )) +
#   geom_point(alpha = 1 / 2) +
#   scale_size(
#     range = c(1, 10),
#     name = "Abundance (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(plottitle," - genus")) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# 
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".genus.", Sys.Date(),".png", sep=""),
#        width = 7, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".genus.", Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

### 4.6 core community-------------------------------------------------------- in manuscript May22

####--- 4.6.1 core community - p - rel. ab.
```{r}
# (physeq.core <- physeq4norm.ansys)
# ## OTU level
# 
# ### OTUs >?% in >?% of samples
# {prev <- 50 # min % of samples
# det <- 0.5 # min % per sample
# (sub.core <- core(physeq.core, detection = det, prevalence = prev/100))
# print(sub.core)}
# mean(rowSums(otu_table(physeq4abs.ansys)))*det/100
# (tax.core <- as.data.frame(sub.core@tax_table@.Data))[-c(1,2,7)]
# unique(tax.core$Class)
# # unique(tax.core$Order)
# tax.core$taxa <- row.names(tax.core)
# # calculate average relative abundance of abundant taxa
# tax.core$av.ab <- colSums(otu_table(physeq.core)[,colnames(otu_table(physeq.core)) %in% tax.core$taxa])/nrow(otu_table(physeq.core)) 
# sum(tax.core$av.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample)
# 
# write.table(tax.core,
#               paste(RESULTS_DIR,"/core.",plottitle,".otu.",det,".",prev,".", Sys.Date(),".txt", sep=""),
#               sep="\t", row.names=FALSE)
# 
# ## genus level
# (physeq.genus <- aggregate_taxa(physeq.core, "Genus"))
# # test <- as.data.frame(physeq.genus@tax_table@.Data)
# 
# ### genera >?% in >?% of samples
# {prev <- 95 # min % of samples
# det <- 0.0 # min % per sample
# (sub.core <- core(physeq.genus, detection = det, prevalence = prev/100))
# print(sub.core)}
# (tax.core <- as.data.frame(sub.core@tax_table@.Data))[-c(1,2,7)]
# # rownames(tax.core) <- NULL
# unique(tax.core$Class)
# 
# # calculate average relative abundance of abundant taxa
# # ab.tax <- as.data.frame(physeq.genus@tax_table@.Data)[as.data.frame(physeq.genus@tax_table@.Data)$Genus %in% tax.core$Genus,] # taxonomy of abundant taxa
# ab.tax.core.mat <- physeq.genus@otu_table@.Data[rownames(physeq.genus@otu_table@.Data) %in% tax.core$Genus,] # creates an OTUmatrix of only the ab.taxonomy
# dim(ab.tax.core.mat)
# # write.table(ab.tax.core.mat,
# #               paste("data.save/abund.tax.mat.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
# #               sep="\t", row.names=FALSE)
# 
# ab.tax.core <- aggregate(rowSums(ab.tax.core.mat), by=list(tax.core$Genus), FUN=sum) # sum of al rel. abundance values grouped per ab.taxa
# ab.tax.core$av.rel.ab <- ab.tax.core$x/ncol(ab.tax.core.mat) # calc average relative abundance per sample
# print(ab.tax.core)
# sum(ab.tax.core$av.rel.ab) # abundant taxa account for X% of total reads (based on rel.ab. per sample) 
# # sum(ds.Ep.mean.norm.OTUs[,colnames(ds.Ep.mean.norm.OTUs) %in% rownames(ab.tax)])/nrow(ds.Ep.mean.norm.OTUs) # X% of total reads
# 
# tax.core.avAb <- merge(tax.core, ab.tax.core[,-2], by.x="Genus", by.y="Group.1")
# sum(tax.core.avAb$av.rel.ab)
# write.table(tax.core.avAb,
#               paste(RESULTS_DIR,"/core.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
#               sep="\t", row.names=FALSE)
```

#### 4.6.2 Venn diagram/ ternary plot  - p - rel. ab. ------------------------ in manuscript May22

##### OTU level - Type
```{r}
physeq.otu4core <- physeq4norm.ansys
catType <- sample_data(physeq.otu4core)$Type
catType <- gsub("0\\.","", catType)
catType <- gsub("1\\.","", catType)
catType <- gsub("2\\.","", catType)

sample_data(physeq.otu4core)$Type <- as.factor(catType)

venn.var <- unique(as.character(meta(physeq.otu4core)$Type))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 50 # min % of samples
det <- 0.5 # min % per sample

for (n in venn.var){ # for each variable n in Types
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq.otu4core, Type == n) # Choose sample from Types by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Types
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}
(abundant.otus <- unique(c(list_core[[1]], list_core[[2]], list_core[[3]])))
(df.abundant <- as.data.frame(physeq.otu4core@tax_table@.Data)[rownames(physeq.otu4core@tax_table) %in% abundant.otus,])


(df.abundant$av.ab <- colSums(otu_table(physeq.otu4core)[,colnames(otu_table(physeq.otu4core)) %in% abundant.otus])/nrow(otu_table(physeq.otu4core)))
(abundant.percOfTotReads <- sum(df.abundant$av.ab)) # % of total reads
```

###### venn.diagram - prevalent OTUs
```{r}
(p <- plot(venn(list_core),
     fills = c(Fp="#d6e2e9", YCp="#cbf3f0", Ep="#48c9c0" ),
     main=paste0(plottitle,"-OTUs-",det,"-", prev)))
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".", Sys.Date(),".png",sep = ""),
       width = 4, height = 3)
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".", Sys.Date(),".svg",sep = ""),
       device = "svg", width = 4, height = 3)

(all.p <- intersect(intersect(list_core$Fp, list_core$YCp), list_core$Ep))
FpYCp <- intersect(list_core$Fp, list_core$YCp)
(FpYCp <- FpYCp[FpYCp %!in% all.p])
YCpEp <- intersect(list_core$YCp, list_core$Ep)
(YCpEp <- YCpEp[YCpEp %!in% all.p])
FpEp <- intersect(list_core$Fp, list_core$Ep)
(FpEp <- FpEp[FpEp %!in% all.p])

(shared <- unique(c(all.p, FpYCp, YCpEp, FpEp)))
length(shared)

(df.shared <- as.data.frame(physeq.otu4core@tax_table@.Data)[rownames(physeq.otu4core@tax_table) %in% shared,])
df.shared$OTU <- rownames(df.shared)

(df.shared$av.ab <- colSums(otu_table(physeq.otu4core)[,colnames(otu_table(physeq.otu4core)) %in% shared])/nrow(otu_table(physeq.otu4core)))
sum(df.shared$av.ab)

unique(df.shared$Class)
(df.all.p <- df.shared[rownames(df.shared) %in% all.p,])
(df.FpYCp <- df.shared[rownames(df.shared) %in% FpYCp,])
(df.YCpEp <- df.shared[rownames(df.shared) %in% YCpEp,])
(df.FpEp <- df.shared[rownames(df.shared) %in% FpEp,])

write.table(df.all.p,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".middle.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.shared,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".shared.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpYCp,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".FpYCp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.YCpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".YCpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".otu.",det,".",prev,".FpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

###### visualizing the shared OTUs as stacked bar charts
```{r}
physeqVennShared <- physeq4norm.ansys
otu_table(physeqVennShared) <- otu_table(physeq4norm.ansys)[,colnames(otu_table(physeq4norm.ansys)) %in% shared]

OTU4tax <- as.data.frame(t(otu_table(physeqVennShared)))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)
dim(OTU4tax)

Taxonomy2 <- cbind(physeqVennShared@tax_table@.Data, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy2[,1:6] <- lapply(Taxonomy2[,1:6], as.factor)
Taxonomy2 <- melt(Taxonomy2, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy2, get_variable(physeqVennShared), by="Name")

ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.4)
```

####### shared OTUs as family taxonomy
```{r}
ab.Taxonomy <- ab.Taxonomy.family

tax.data2plot0 <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Type = ab.Taxonomy$Type, Ant = ab.Taxonomy$Ant, Age = ab.Taxonomy$Age, Tree = ab.Taxonomy$TreeR)

# exclude samples
tax.data2plot <- tax.data2plot0#[-grep("unknown", tax.data2plot0$Type), ]

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "Type", "Ant", "Age", "Tree"), summarise, Freq = sum(Freq))

# removing "_unclassified"
tax.data2plot$Taxa <- gsub("_unclassified","_uncl", tax.data2plot$Taxa)

# for stacked bar charts
  ## f?gt relative abundance f?r OTU innerhalb "Sample" (alle Cecropia samples)
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per sample
AllSample.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa,main=plottitle) + scale_fill_manual(values=colPalette03) +
    scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1))
AllSample.bar

ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family",Sys.Date(),"png",sep = "."),width = 9, height = 7, dpi = 300)
ggsave(paste(PLOTS_DIR,"/bar.tax.Epmean.family",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 9, height = 7, dpi = 300)

# AllSample.bar + facet_wrap( ~ Ant*Type, scale="free", ncol=3) # rel. Abundance
```

##### OTU level - SuccessionType --------------------------------------------- in manuscript May22
```{r}
# checking for shared prevalent OTUs between IP/YP/EP should take the ant species in EPs into account.
# as EP community composition differs significantly between ant species (alf|con), there are only a few prevalent EP OTUs.
# the idea is to calculate prevalent OTUs of EP_alf and EP_con
# and to merge [unique(c(,))] the prevalent OTUs of EP_alf and EP_con to "EP" for comparon with IP and YP 

physeq.otu4core <- physeq4norm.ansys
sample_data(physeq.otu4core)$Succ <- gsub("0\\.","", sample_data(physeq.otu4core)$Succ)
sample_data(physeq.otu4core)$Succ <- gsub("1\\.","", sample_data(physeq.otu4core)$Succ)
sample_data(physeq.otu4core)$Succ <- as.factor(gsub("2\\.","", sample_data(physeq.otu4core)$Succ))

venn.var <- unique(as.character(meta(physeq.otu4core)$Succ))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 50 # min % of samples
det <- 0.5 # min % per sample

for (n in venn.var){ # for each variable n in SuccessionTypes
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq.otu4core, Succ == n) # Choose sample from SuccessionTypes by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from SuccessionType n 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each SuccessionTypes
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}
(abundant.otus <- unique(c(list_core[[1]], list_core[[2]], list_core[[3]], list_core[[4]])))
df.abundant <- as.data.frame(physeq.otu4core@tax_table@.Data)[rownames(physeq.otu4core@tax_table) %in% abundant.otus,]
df.abundant$OTU <- rownames(df.abundant)
df.abundant$av.ab <- colSums(otu_table(physeq.otu4core)[,colnames(otu_table(physeq.otu4core)) %in% abundant.otus])/nrow(otu_table(physeq.otu4core))
df.abundant
(abundant.percOfTotReads <- sum(df.abundant$av.ab)) # % of total reads
df.abundant.sort <- df.abundant[order(df.abundant$Phylum, df.abundant$Class, df.abundant$Order, df.abundant$Family, df.abundant$Genus),]
df.abundant.sort
```

###### venn.diagram - prevalent OTUs (separated & merged EP_alf, EP_con) ----- in manuscript May22
```{r}
(p <- plot(venn(list_core),
     fills = c(Fp="#d6e2e9", YCp="#cbf3f0", Ep="#48c9c0" ),
     main=paste0(plottitle,"-OTUs-",det,"-", prev)))
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".", Sys.Date(),".png",sep = ""),
       width = 4, height = 3)
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".", Sys.Date(),".svg",sep = ""),
       device = "svg", width = 4, height = 3)

# # sharing OTUs IP, YP, EP (prevalent OTUs of EP_alf & EP_con)
# FpYCpEp_alf <- intersect(intersect(list_core$Fp, list_core$YCp),list_core$Ep_alf)
# FpYCpEp_con <- intersect(intersect(list_core$Fp, list_core$YCp), list_core$Ep_con)
# (all.p <- unique(c(FpYCpEp_alf, FpYCpEp_con)))
# 
# FpYCp <- intersect(list_core$Fp, list_core$YCp)
# (FpYCp <- FpYCp[FpYCp %!in% all.p])
# 
# YCpEp_alf <- intersect(list_core$YCp, list_core$Ep_alf)
# YCpEp_con <- intersect(list_core$YCp, list_core$Ep_con)
# YCpEp <- c(YCpEp_alf, YCpEp_con)
# (YCpEp <- YCpEp[YCpEp %!in% all.p])
# 
# FpEp <- intersect(list_core$Fp, list_core$Ep)
# (FpEp <- FpEp[FpEp %!in% all.p])

# alternative way by merging EP_alf & EP_con early
list_core$Ep <- unique(c(list_core$Ep_alf,list_core$Ep_con))
list_core$Ep_alf <- NULL
list_core$Ep_con <- NULL

(p <- plot(venn(list_core),
     fills = c(Fp="#d6e2e9", YCp="#cbf3f0", Ep="#48c9c0" ),
     main=paste0(plottitle,"-OTUs-",det,"-", prev)))
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.mergedEP.otu.",det,".",prev,".", Sys.Date(),".png",sep = ""),
       width = 4, height = 3)
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.mergedEP.otu.",det,".",prev,".", Sys.Date(),".svg",sep = ""),
       device = "svg", width = 4, height = 3)

(all.p <- intersect(intersect(list_core$Fp, list_core$YCp), list_core$Ep))
FpYCp <- intersect(list_core$Fp, list_core$YCp)
(FpYCp <- FpYCp[FpYCp %!in% all.p])
YCpEp <- intersect(list_core$YCp, list_core$Ep)
(YCpEp <- YCpEp[YCpEp %!in% all.p])
FpEp <- intersect(list_core$Fp, list_core$Ep)
(FpEp <- FpEp[FpEp %!in% all.p])

(shared <- unique(c(all.p, FpYCp, YCpEp, FpEp)))
length(shared)

(EpOnly <- list_core$Ep[list_core$Ep %!in% shared])
(YPOnly <- list_core$YCp[list_core$YCp %!in% shared])
(FpOnly <- list_core$Fp[list_core$Fp %!in% shared])

# (df.shared <- as.data.frame(physeq.otu4core@tax_table@.Data)[rownames(physeq.otu4core@tax_table) %in% shared,])
# df.shared$OTU <- rownames(df.shared)
# (df.shared$av.ab <- colSums(otu_table(physeq.otu4core)[,colnames(otu_table(physeq.otu4core)) %in% shared])/nrow(otu_table(physeq.otu4core)))
# sum(df.shared$av.ab)
# df.shared <- df.shared[order(df.shared$Phylum, df.shared$Class, df.shared$Order, df.shared$Family, df.shared$Genus),]

unique(df.abundant.sort$Class)
(df.all.p <- df.abundant.sort[rownames(df.abundant.sort) %in% all.p,])
(df.FpYCp <- df.abundant.sort[rownames(df.abundant.sort) %in% FpYCp,])
(df.YCpEp <- df.abundant.sort[rownames(df.abundant.sort) %in% YCpEp,])
(df.FpEp <- df.abundant.sort[rownames(df.abundant.sort) %in% FpEp,])
(df.shared <- df.abundant.sort[rownames(df.abundant.sort) %in% shared,])
sum(df.shared$av.ab)
(df.Ep <- df.abundant.sort[rownames(df.abundant.sort) %in% EpOnly,])
(df.Yp <- df.abundant.sort[rownames(df.abundant.sort) %in% YPOnly,])
(df.Fp <- df.abundant.sort[rownames(df.abundant.sort) %in% FpOnly,])


write.table(df.all.p,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".middle.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.shared,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".shared.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpYCp,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".FpYCp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.YCpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".YCpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".FpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)

df.allPrevalent <- cbind(
    # "OTU" = c(rownames(df.all.p), rownames(df.FpYCp), rownames(df.YCpEp), rownames(df.FpEp), rownames(df.Fp), rownames(df.Yp), rownames(df.Ep)),
    "category" =  c(rep("All",nrow(df.all.p)), 
                    rep("IP_YP",nrow(df.FpYCp)), 
                    rep("YP_EP",nrow(df.YCpEp)), 
                    rep("IP_EP",nrow(df.FpEp)), 
                    rep("IP",nrow(df.Fp)), 
                    rep("YP",nrow(df.Yp)),
                    rep("EP",nrow(df.Ep))
                    ),
    rbind(df.all.p, 
          df.FpYCp, 
          df.YCpEp, 
          df.FpEp, 
          df.Fp, 
          df.Yp, 
          df.Ep)
    )
write.table(df.allPrevalent,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.otu.",det,".",prev,".allPrevalent.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

###### ternary.plot - prevalent OTUs
```{r}
physeq.otu4core.tern <- physeq.otu4core

# remove 0./1./2. from column names
sample_data(physeq.otu4core.tern)$Type <- gsub("0\\.","", sample_data(physeq.otu4core.tern)$Type)
sample_data(physeq.otu4core.tern)$Type <- gsub("1\\.","", sample_data(physeq.otu4core.tern)$Type)
sample_data(physeq.otu4core.tern)$Type <- as.factor(gsub("2\\.","", sample_data(physeq.otu4core.tern)$Type))

# merge samples to 4 SuccTypes 
(physeq.otu4core_Succ <- merge_samples(physeq.otu4core.tern, "Succ", fun = mean)) # merge by source

rowSums(otu_table(physeq.otu4core_Succ))

# calc relative abundance per SuccType
(physeq.otu4core_Succ.rel <- transform_sample_counts(physeq.otu4core_Succ, function(x) x / sum(x) * 100 )) # rel abundance per type

# replace 1,2,3 after merging with the respective Types again 
sample_data(physeq.otu4core_Succ.rel)$Type <- rownames(sample_data(physeq.otu4core_Succ.rel))
sample_data(physeq.otu4core_Succ.rel)$Type <- gsub("_.*","", sample_data(physeq.otu4core_Succ.rel)$Type)

# mOm: mean of means... merging the average EP_alf with the average EP_con community, to not get a biased community as EP_con is overrepresented 
(physeq.otu4core_SuccType <- merge_samples(physeq.otu4core_Succ.rel, "Type", fun = mean)) # merge by source
# calc new relative abundance
(physeq.otu4core_SuccType.rel <- transform_sample_counts(physeq.otu4core_SuccType, function(x) x / sum(x) * 100 )) # rel abundance per type

# ternary plot of only prevalent OTUs (IP, YP, EP [EP_alf or EP_con])
## there is no bias along different number of samples!

(physeq4tern.core <-
    subset_taxa(physeq.otu4core_SuccType.rel, rownames(tax_table(physeq.otu4core_SuccType.rel)) %in% abundant.otus))

## plot abundance of only prevalent OTU reads

# meandf <- as(otu_table(physeq4tern.core), "matrix")
# if (!taxa_are_rows(physeq4tern.core)) { meandf <- t(meandf) }
# abundance <- rowSums(meandf) / sum(meandf) * 100
# p.core.4Ternary.abundantReads <- data.frame(
#   meandf,
#   Abundance = abundance,
#   Phylum = tax_table(physeq4tern.core)[, "Phylum"],
#   Class = tax_table(physeq4tern.core)[, "Class"],
#   Order = tax_table(physeq4tern.core)[, "Order"],
#   Genus = tax_table(physeq4tern.core)[, "Genus"]
# )

## plot abundance of total reads

meandf <- as(otu_table(physeq.otu4core_SuccType.rel), "matrix")
if (!taxa_are_rows(physeq.otu4core_SuccType.rel)) { meandf <- t(meandf) }
abundance <- rowSums(meandf) / sum(meandf) * 100

p.all.meanAb <- data.frame(
  meandf,
  Abundance = abundance)

p.core.meanAb <- p.all.meanAb[rownames(p.all.meanAb) %in% abundant.otus,]

p.core.4Ternary.totalReads <- data.frame(
  p.core.meanAb,
  Phylum = tax_table(physeq4tern.core)[, "Phylum"],
  Class = tax_table(physeq4tern.core)[, "Class"],
  Order = tax_table(physeq4tern.core)[, "Order"],
  Genus = tax_table(physeq4tern.core)[, "Genus"],
  OTU = paste0(rownames(tax_table(physeq4tern.core)),"_",Genus = tax_table(physeq4tern.core)[, "Genus"])
)

write.table(p.core.4Ternary.totalReads,
            paste(RESULTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.otus),".core.otus.",det,".",prev,".genus.",
                  Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)

# ## plot ternary - only abundant OTUs
# # color = Genus
# df4Ternary <- p.core.4Ternary.totalReads # p.core.4Ternary.abundantReads # p.core.4Ternary.totalReads
# p_ternary_p <-
#   ggtern(data = df4Ternary,
#          aes(
#            x = Fp,
#            y = YCp,
#            z = Ep,
#            size = Abundance,
#            colour = Genus # OTU/Genus
#          )) +
#   geom_point(alpha = 3 / 6) +
#   scale_size(
#     range = c(1, 10),
#     name = "mean rel. abundance across all types (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(
#     plottitle,"-",length(abundant.otus),"-OTUs-",det,"-", prev, " % of total reads ",round(abundant.percOfTotReads,2))) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.otus),".core.otus.",det,".",prev,".genus.",
#              Sys.Date(),".png", sep=""),
#        width = 8, height = 7, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.otus),"core.otus.",det,".",prev,".genus.",
#              Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 8, height = 7, dpi = 300)
# 
# # color = Order
# df4Ternary <- p.core.4Ternary.totalReads # p.core.4Ternary.abundantReads # p.core.4Ternary.totalReads
# p_ternary_p <-
#   ggtern(data = df4Ternary,
#          aes(
#            x = Fp,
#            y = YCp,
#            z = Ep,
#            size = Abundance,
#            colour = Order # Genus
#          )) +
#   geom_point(alpha = 3 / 6) +
#   scale_size(
#     range = c(1, 10),
#     name = "mean rel. abundance across all types (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(
#     plottitle,"-",length(abundant.otus),"-OTUs-",det,"-", prev, " % of total reads ",round(abundant.percOfTotReads,2))) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.otus),".core.otus.",det,".",prev,".order.",
#              Sys.Date(),".png", sep=""),
#        width = 8, height = 7, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.otus),"core.otus.",det,".",prev,".order.",
#              Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 8, height = 7, dpi = 300)
```

##### genus level - SuccType - new
```{r}
(physeq.genus4core <- aggregate_taxa(physeq4norm.ansys, "Genus"))

sample_data(physeq.genus4core)$Succ <- gsub("0\\.","", sample_data(physeq.genus4core)$Succ)
sample_data(physeq.genus4core)$Succ <- gsub("1\\.","", sample_data(physeq.genus4core)$Succ)
sample_data(physeq.genus4core)$Succ <- as.factor(gsub("2\\.","", sample_data(physeq.genus4core)$Succ))

venn.var <- unique(as.character(meta(physeq.genus4core)$Succ))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 99 # min % of samples
det <- 0.0 # min % per sample

for (n in venn.var){ # for each variable n in SuccessionTypes
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq.genus4core, Succ == n) # Choose sample from SuccessionTypes by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from SuccessionType n 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each SuccessionTypes
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}
(abundant.tax <- unique(c(list_core[[1]], list_core[[2]], list_core[[3]], list_core[[4]])))
df.abundant <- as.data.frame(physeq.genus4core@tax_table@.Data)[rownames(physeq.genus4core@tax_table) %in% abundant.tax,]
# df.abundant$OTU <- rownames(df.abundant)
df.abundant$av.ab <- rowSums(otu_table(physeq.genus4core)[rownames(otu_table(physeq.genus4core)) %in% abundant.tax,])/ncol(otu_table(physeq.genus4core))
df.abundant
(abundant.percOfTotReads <- sum(df.abundant$av.ab)) # % of total reads
df.abundant.sort <- df.abundant[order(df.abundant$Phylum, df.abundant$Class, df.abundant$Order, df.abundant$Family, df.abundant$Genus),]
df.abundant.sort
```

###### venn.diagram - prevalent genera (separated & merged EP_alf, EP_con) ----- in manuscript May22
```{r}
(p <- plot(venn(list_core),
     fills = c(Fp="#d6e2e9", YCp="#cbf3f0", Ep="#48c9c0" ),
     main=paste0(plottitle,"-OTUs-",det,"-", prev)))
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".", Sys.Date(),".png",sep = ""),
       width = 4, height = 3)
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".", Sys.Date(),".svg",sep = ""),
       device = "svg", width = 4, height = 3)

# # sharing OTUs IP, YP, EP (prevalent OTUs of EP_alf & EP_con)
# FpYCpEp_alf <- intersect(intersect(list_core$Fp, list_core$YCp),list_core$Ep_alf)
# FpYCpEp_con <- intersect(intersect(list_core$Fp, list_core$YCp), list_core$Ep_con)
# (all.p <- unique(c(FpYCpEp_alf, FpYCpEp_con)))
# 
# FpYCp <- intersect(list_core$Fp, list_core$YCp)
# (FpYCp <- FpYCp[FpYCp %!in% all.p])
# 
# YCpEp_alf <- intersect(list_core$YCp, list_core$Ep_alf)
# YCpEp_con <- intersect(list_core$YCp, list_core$Ep_con)
# YCpEp <- c(YCpEp_alf, YCpEp_con)
# (YCpEp <- YCpEp[YCpEp %!in% all.p])
# 
# FpEp <- intersect(list_core$Fp, list_core$Ep)
# (FpEp <- FpEp[FpEp %!in% all.p])

# alternative way by merging EP_alf & EP_con early
list_core$Ep <- unique(c(list_core$Ep_alf,list_core$Ep_con))
list_core$Ep_alf <- NULL
list_core$Ep_con <- NULL

(p <- plot(venn(list_core),
     fills = c(Fp="#d6e2e9", YCp="#cbf3f0", Ep="#48c9c0" ),
     main=paste0(plottitle,"-OTUs-",det,"-", prev)))
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.mergedEP.genus.",det,".",prev,".", Sys.Date(),".png",sep = ""),
       width = 4, height = 3)
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".Succ.mergedEP.genus.",det,".",prev,".", Sys.Date(),".svg",sep = ""),
       device = "svg", width = 4, height = 3)

(all.p <- intersect(intersect(list_core$Fp, list_core$YCp), list_core$Ep))
FpYCp <- intersect(list_core$Fp, list_core$YCp)
(FpYCp <- FpYCp[FpYCp %!in% all.p])
YCpEp <- intersect(list_core$YCp, list_core$Ep)
(YCpEp <- YCpEp[YCpEp %!in% all.p])
FpEp <- intersect(list_core$Fp, list_core$Ep)
(FpEp <- FpEp[FpEp %!in% all.p])

(shared <- unique(c(all.p, FpYCp, YCpEp, FpEp)))
length(shared)

(EpOnly <- list_core$Ep[list_core$Ep %!in% shared])
(YPOnly <- list_core$YCp[list_core$YCp %!in% shared])
(FpOnly <- list_core$Fp[list_core$Fp %!in% shared])

unique(df.abundant.sort$Class)
(df.all.p <- df.abundant.sort[rownames(df.abundant.sort) %in% all.p,])
(df.FpYCp <- df.abundant.sort[rownames(df.abundant.sort) %in% FpYCp,])
(df.YCpEp <- df.abundant.sort[rownames(df.abundant.sort) %in% YCpEp,])
(df.FpEp <- df.abundant.sort[rownames(df.abundant.sort) %in% FpEp,])
(df.shared <- df.abundant.sort[rownames(df.abundant.sort) %in% shared,])
sum(df.shared$av.ab)
(df.Ep <- df.abundant.sort[rownames(df.abundant.sort) %in% EpOnly,])
(df.Yp <- df.abundant.sort[rownames(df.abundant.sort) %in% YPOnly,])
(df.Fp <- df.abundant.sort[rownames(df.abundant.sort) %in% FpOnly,])


write.table(df.all.p,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".middle.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.shared,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".shared.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpYCp,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".FpYCp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.YCpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".YCpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".FpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)

df.allPrevalent <- cbind(
    # "OTU" = c(rownames(df.all.p), rownames(df.FpYCp), rownames(df.YCpEp), rownames(df.FpEp), rownames(df.Fp), rownames(df.Yp), rownames(df.Ep)),
    "category" =  c(rep("All",nrow(df.all.p)), 
                    rep("IP_YP",nrow(df.FpYCp)), 
                    rep("YP_EP",nrow(df.YCpEp)), 
                    rep("IP_EP",nrow(df.FpEp)), 
                    rep("IP",nrow(df.Fp)), 
                    rep("YP",nrow(df.Yp)),
                    rep("EP",nrow(df.Ep))
                    ),
    rbind(df.all.p, 
          df.FpYCp, 
          df.YCpEp, 
          df.FpEp, 
          df.Fp, 
          df.Yp, 
          df.Ep)
    )
write.table(df.allPrevalent,
            paste(RESULTS_DIR,"/venn.",plottitle,".Succ.genus.",det,".",prev,".allPrevalent.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

###### ternary.plot - prevalent genera
```{r}
physeq.genus4core.tern <- physeq.genus4core

# remove 0./1./2. from column names
sample_data(physeq.genus4core.tern)$Type <- gsub("0\\.","", sample_data(physeq.genus4core.tern)$Type)
sample_data(physeq.genus4core.tern)$Type <- gsub("1\\.","", sample_data(physeq.genus4core.tern)$Type)
sample_data(physeq.genus4core.tern)$Type <- as.factor(gsub("2\\.","", sample_data(physeq.genus4core.tern)$Type))

# merge samples to 4 SuccTypes 
(physeq.genus4core_Succ <- merge_samples(physeq.genus4core.tern, "Succ", fun = mean)) # merge by source

rowSums(otu_table(physeq.genus4core_Succ))

# calc relative abundance per SuccType
(physeq.genus4core_Succ.rel <- transform_sample_counts(physeq.genus4core_Succ, function(x) x / sum(x) * 100 )) # rel abundance per type

# replace 1,2,3 after merging with the respective Types again 
sample_data(physeq.genus4core_Succ.rel)$Type <- rownames(sample_data(physeq.genus4core_Succ.rel))
sample_data(physeq.genus4core_Succ.rel)$Type <- gsub("_.*","", sample_data(physeq.genus4core_Succ.rel)$Type)

# mOm: mean of means... merging the average EP_alf with the average EP_con community, to not get a biased community as EP_con is overrepresented 
(physeq.genus4core_SuccType <- merge_samples(physeq.genus4core_Succ.rel, "Type", fun = mean)) # merge by source
# calc new relative abundance
(physeq.genus4core_SuccType.rel <- transform_sample_counts(physeq.genus4core_SuccType, function(x) x / sum(x) * 100 )) # rel abundance per type

# ternary plot of only prevalent OTUs (IP, YP, EP [EP_alf or EP_con])
## there is no bias along different number of samples!

(physeq4tern.core <-
    subset_taxa(physeq.genus4core_SuccType.rel, rownames(tax_table(physeq.genus4core_SuccType.rel)) %in% abundant.tax))

## plot abundance of only prevalent OTU reads

# meandf <- as(otu_table(physeq4tern.core), "matrix")
# if (!taxa_are_rows(physeq4tern.core)) { meandf <- t(meandf) }
# abundance <- rowSums(meandf) / sum(meandf) * 100
# p.core.4Ternary.abundantReads <- data.frame(
#   meandf,
#   Abundance = abundance,
#   Phylum = tax_table(physeq4tern.core)[, "Phylum"],
#   Class = tax_table(physeq4tern.core)[, "Class"],
#   Order = tax_table(physeq4tern.core)[, "Order"],
#   Genus = tax_table(physeq4tern.core)[, "Genus"]
# )

## plot abundance of total reads

meandf <- as(otu_table(physeq.genus4core_SuccType.rel), "matrix")
if (!taxa_are_rows(physeq.genus4core_SuccType.rel)) { meandf <- t(meandf) }
abundance <- rowSums(meandf) / sum(meandf) * 100

p.all.meanAb <- data.frame(
  meandf,
  Abundance = abundance)

p.core.meanAb <- p.all.meanAb[rownames(p.all.meanAb) %in% abundant.tax,]

p.core.4Ternary.totalReads <- data.frame(
  p.core.meanAb,
  Phylum = tax_table(physeq4tern.core)[, "Phylum"],
  Class = tax_table(physeq4tern.core)[, "Class"],
  Order = tax_table(physeq4tern.core)[, "Order"],
  Genus = tax_table(physeq4tern.core)[, "Genus"],
  OTU = paste0(rownames(tax_table(physeq4tern.core)),"_",Genus = tax_table(physeq4tern.core)[, "Genus"])
)

write.table(p.core.4Ternary.totalReads,
            paste(RESULTS_DIR,"/ternary.",plottitle,".Succ.genus.",length(abundant.tax),".core.",det,".",prev,".genus.",
                  Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)

# ## plot ternary - only abundant OTUs
# # color = Genus
# df4Ternary <- p.core.4Ternary.totalReads # p.core.4Ternary.abundantReads # p.core.4Ternary.totalReads
# p_ternary_p <-
#   ggtern(data = df4Ternary,
#          aes(
#            x = Fp,
#            y = YCp,
#            z = Ep,
#            size = Abundance,
#            colour = Genus # OTU/Genus
#          )) +
#   geom_point(alpha = 3 / 6) +
#   scale_size(
#     range = c(1, 10),
#     name = "mean rel. abundance across all types (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(
#     plottitle,"-",length(abundant.tax),"-OTUs-",det,"-", prev, " % of total reads ",round(abundant.percOfTotReads,2))) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.tax),".core.otus.",det,".",prev,".genus.",
#              Sys.Date(),".png", sep=""),
#        width = 8, height = 7, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.tax),"core.otus.",det,".",prev,".genus.",
#              Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 8, height = 7, dpi = 300)
# 
# # color = Order
# df4Ternary <- p.core.4Ternary.totalReads # p.core.4Ternary.abundantReads # p.core.4Ternary.totalReads
# p_ternary_p <-
#   ggtern(data = df4Ternary,
#          aes(
#            x = Fp,
#            y = YCp,
#            z = Ep,
#            size = Abundance,
#            colour = Order # Genus
#          )) +
#   geom_point(alpha = 3 / 6) +
#   scale_size(
#     range = c(1, 10),
#     name = "mean rel. abundance across all types (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(
#     plottitle,"-",length(abundant.tax),"-OTUs-",det,"-", prev, " % of total reads ",round(abundant.percOfTotReads,2))) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.tax),".core.otus.",det,".",prev,".order.",
#              Sys.Date(),".png", sep=""),
#        width = 8, height = 7, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".Succ.",length(abundant.tax),"core.otus.",det,".",prev,".order.",
#              Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 8, height = 7, dpi = 300)
```

##### genus level - old
```{r}
(physeq.genus4core <- aggregate_taxa(physeq4norm.ansys, "Genus"))
catType <- sample_data(physeq.genus4core)$Type
catType <- gsub("0\\.","", catType)
catType <- gsub("1\\.","", catType)
catType <- gsub("2\\.","", catType)

sample_data(physeq.genus4core)$Type <- as.factor(catType)

venn.var <- unique(as.character(meta(physeq.genus4core)$Type))
print(venn.var)

list_core <- c() # an empty object to store information

prev <- 50 # min % of samples
det <- 1.0 # min % per sample

for (n in venn.var){ # for each variable n in Types
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(physeq.genus4core, Type == n) # Choose sample from Types by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = det, 
                           prevalence = prev/100)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each Types
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    # print(list_core)
}
(abundant.genera <- unique(c(list_core[[1]], list_core[[2]], list_core[[3]])))
(df.abundant <- as.data.frame(physeq.genus4core@tax_table@.Data)[physeq.genus4core@tax_table@.Data[,"Genus"] %in% abundant.genera,])

(df.abundant$av.ab <- rowSums(otu_table(physeq.genus4core)[rownames(otu_table(physeq.genus4core)) %in% abundant.genera,])/ncol(otu_table(physeq.genus4core)))
(abundant.percOfTotReads <- sum(df.abundant$av.ab)) # % of total reads
nrow(df.abundant) # no. of genera
```

###### venn.diagram - prevalent genera
```{r}
(p <- plot(venn(list_core),
     fills = c(Fp="#d6e2e9", YCp="#cbf3f0", Ep="#48c9c0" ),
     main=paste0(plottitle,"-genera-",det,"-", prev)))
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".png",sep = ""),
       width = 4, height = 3)
ggsave(p, file=paste(PLOTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".svg",sep = ""), 
       device = "svg", width = 4, height = 3)

(all.p <- intersect(intersect(list_core$Fp, list_core$YCp), list_core$Ep))
FpYCp <- intersect(list_core$Fp, list_core$YCp)
(FpYCp <- FpYCp[FpYCp %!in% all.p])
YCpEp <- intersect(list_core$YCp, list_core$Ep)
(YCpEp <- YCpEp[YCpEp %!in% all.p])
FpEp <- intersect(list_core$Fp, list_core$Ep)
(FpEp <- FpEp[FpEp %!in% all.p])

(shared <- unique(c(all.p, FpYCp, YCpEp, FpEp)))
length(shared)

sq <- seq(max(length(all.p), length(FpYCp), length(YCpEp), length(FpEp)))
d9 <- data.frame(all.p[sq], FpYCp[sq], YCpEp[sq], FpEp[sq])
colnames(d9) <- c("all.p","FpYCp","YCpEp", "FpEp") # per site
print(d9)
write.table(d9,
              paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
              sep="\t", row.names=FALSE)

# (av.ab <- rowSums(otu_table(physeq.genus4core)[rownames(otu_table(physeq.genus4core)) %in% shared,])/ncol(otu_table(physeq.genus4core)))
# sum(av.ab)
# write.table(av.ab,
#               paste(RESULTS_DIR,"/venn.",plottitle,".unique.genus.",det,".",prev,".", Sys.Date(),".txt", sep=""),
#               sep="\t", row.names=TRUE)

(df.shared <- as.data.frame(physeq.genus4core@tax_table@.Data)[(physeq.genus4core@tax_table@.Data)[,"Genus"] %in% shared,])

(df.shared$av.ab <- rowSums(otu_table(physeq.genus4core)[(physeq.genus4core@tax_table@.Data)[,"Genus"] %in% shared,])/ncol(otu_table(physeq.genus4core)))
sum(df.shared$av.ab)
print(df.shared)
unique(df.shared$Class)
write.table(df.shared,
            paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".shared.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)

(df.all.p <- df.shared[rownames(df.shared) %in% all.p,])
(df.FpYCp <- df.shared[rownames(df.shared) %in% FpYCp,])
(df.YCpEp <- df.shared[rownames(df.shared) %in% YCpEp,])
(df.FpEp <- df.shared[rownames(df.shared) %in% FpEp,])

write.table(df.all.p,
            paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".middle.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpYCp,
            paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".FpYCp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.YCpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".YCpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
write.table(df.FpEp,
            paste(RESULTS_DIR,"/venn.",plottitle,".genus.",det,".",prev,".FpEp.", Sys.Date(),".txt", sep=""),
            sep="\t", row.names=FALSE)
```

###### ternary.plot - prevalent genera
```{r}
# (physeq.genus4core_merged <- merge_samples(physeq.genus4core, "Type", fun = mean)) # merge by type
# (physeq.genus4core_merged_rel <- transform_sample_counts(physeq.genus4core_merged, function(x) x / sum(x) * 100 )) # rel abundance per type
# 
# # only abundant genera
# 
# (physeq4tern.core <- 
#     subset_taxa(physeq.genus4core_merged_rel, rownames(tax_table(physeq.genus4core_merged_rel)) %in% abundant.genera))
# 
# ## plot abundance of total reads
# 
# meandf <- as(otu_table(physeq.genus4core_merged_rel), "matrix")
# if (!taxa_are_rows(physeq.genus4core_merged_rel)) { meandf <- t(meandf) }
# abundance <- rowSums(meandf) / sum(meandf) * 100
# 
# p.all.meanAb <- data.frame(
#   meandf,
#   Abundance = abundance)
# 
# p.core.meanAb <- p.all.meanAb[rownames(p.all.meanAb) %in% abundant.genera,]
# 
# p.core.4Ternary.totalReads <- data.frame(
#   p.core.meanAb,
#   Phylum = tax_table(physeq4tern.core)[, "Phylum"],
#   Class = tax_table(physeq4tern.core)[, "Class"],
#   Order = tax_table(physeq4tern.core)[, "Order"],
#   Genus = tax_table(physeq4tern.core)[, "Genus"]
# )
# 
# # color = order
# df4Ternary <- p.core.4Ternary.totalReads 
# p_ternary_p <-
#   ggtern(data = df4Ternary, 
#          aes(
#            x = Fp,
#            y = YCp,
#            z = Ep,
#            size = Abundance,
#            colour = Order # Order, Genus
#          )) +
#   geom_point(alpha = 3 / 6) +
#   scale_size(
#     range = c(1, 10),
#     name = "mean rel. abundance across all types (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(plottitle,"-",length(abundant.genera),"-genera-",det,"-", prev, " % of total reads ",round(abundant.percOfTotReads,2))) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".core.genera.",det,".",prev,".order.", Sys.Date(),".png", sep=""),
#        width = 8, height = 7, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".core.genera.",det,".",prev,".order.", Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 8, height = 7, dpi = 300)
# 
# # color = genus
# df4Ternary <- p.core.4Ternary.totalReads 
# p_ternary_p <-
#   ggtern(data = df4Ternary, 
#          aes(
#            x = Fp,
#            y = YCp,
#            z = Ep,
#            size = Abundance,
#            colour = Genus # Order, Genus
#          )) +
#   geom_point(alpha = 3 / 6) +
#   scale_size(
#     range = c(1, 10),
#     name = "mean rel. abundance across all types (%)"
#   ) +
#   theme_arrownormal() +
#     scale_color_manual(values = colPalette01) +
#   guides(colour = guide_legend(override.aes = list(size = 3))) +
#   ggtitle(paste0(plottitle,"-",length(abundant.genera),"-genera-",det,"-", prev, " % of total reads ",round(abundant.percOfTotReads,2))) +
#   theme(axis.title = element_blank())
# print(p_ternary_p)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".core.genera.",det,".",prev,".genus.", Sys.Date(),".png", sep=""),
#        width = 8, height = 7, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/ternary.",plottitle,".core.genera.",det,".",prev,".genus.", Sys.Date(),".pdf", sep=""),
#        useDingbats=FALSE,width = 8, height = 7, dpi = 300)
```




## 5. succession within ant sp.

### 5.1 alfari
```{r}
(physeq4abs.ansys <- physeq.p.alf.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.p.alf.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.p.alf.newEpmean"

table(get_variable(physeq4norm.ansys)$Ant)
sum(table(get_variable(physeq4norm.ansys)$Ant))
length(unique(paste0(get_variable(physeq4norm.ansys)$Tree)))
```

#### 5.1.1 alpha diversity - shannon
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

plot_richness(physeq4abs.ansys, measures = c("Shannon"))

ggsave(paste(PLOTS_DIR,"/alpha.plot.p.alf",Sys.Date(),"png",sep = "."),width = 10, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.plot.p.alf",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 10, height = 6, dpi = 300)

### testing patch type alpha diversity for significance
alpha.alf <- get_variable(physeq4abs.ansys)
alpha.alf$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.alf.res <- kruskal.test(Shannon ~ Type, data = alpha.alf))
(alpha.alf.res.pw <- pairwise.wilcox.test(alpha.alf$Shannon, alpha.alf$Type, p.adjust.method = "BH"))

alpha.alf.sub <- alpha.alf[grep("0.Fp", alpha.alf$Type),]
median(alpha.alf.sub$Shannon)
alpha.alf.sub <- alpha.alf[grep("1.YCp", alpha.alf$Type),]
median(alpha.alf.sub$Shannon)
alpha.alf.sub <- alpha.alf[grep("2.Ep", alpha.alf$Type),]
median(alpha.alf.sub$Shannon)

p <- ggplot(alpha.alf, aes(Type, Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.alf.dots",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.alf.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)

p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  # geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.alf",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.alf",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)
```

#### 5.1.2 beta diversity - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.alf.Type",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.alf.Type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)

# Type & Ant
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Type", variance = TRUE, connect = FALSE)

# Type & Type
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.alf.TypeType",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.alf.TypeType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

#### 5.1.3 drivers

##### adonis
```{r}
(m.p.Type <- adonis(otu_table(physeq4norm.ansys) ~ Type,
                    data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

##### plotting - dbRDA
```{r}
dbRDA.obj <- capscale((otu_table(physeq4norm.ansys)) ~ Type, data=get_variable(physeq4norm.ansys),  distance = "bray")

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.p.alf.Type",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.p.alf.Type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```

### 5.2 constructor
```{r}
(physeq4abs.ansys <- physeq.p.con.abs)
(physeq4abs.ansys <- prune_taxa(taxa_sums(physeq4abs.ansys) > 0, physeq4abs.ansys))
(physeq4norm.ansys <- physeq.p.con.norm)
(physeq4norm.ansys <- prune_taxa(taxa_sums(physeq4norm.ansys) > 0, physeq4norm.ansys))
plottitle <- "ds.p.con.newEpmean"

table(get_variable(physeq4norm.ansys)$Ant)
sum(table(get_variable(physeq4norm.ansys)$Ant))
length(unique(paste0(get_variable(physeq4norm.ansys)$Tree)))
```

#### 5.2.1 alpha diversity - shannon
```{r}
### Alpha diversität - phyloseq - not rarefy/transform beforehand!

plot_richness(physeq4abs.ansys, measures = c("Shannon"))

ggsave(paste(PLOTS_DIR,"/alpha.plot.p.con",Sys.Date(),"png",sep = "."),width = 10, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.plot.p.con",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 10, height = 6, dpi = 300)

### testing patch type alpha diversity for significance
alpha.con <- get_variable(physeq4abs.ansys)
alpha.con$Shannon <- estimate_richness(physeq4abs.ansys, measures = c("Shannon"))$Shannon

(alpha.con.res <- kruskal.test(Shannon ~ Type, data = alpha.con))
(alpha.con.res.pw <- pairwise.wilcox.test(alpha.con$Shannon, alpha.con$Type, p.adjust.method = "BH"))

alpha.con.sub <- alpha.con[grep("0.Fp", alpha.con$Type),]
median(alpha.con.sub$Shannon)
alpha.con.sub <- alpha.con[grep("1.YCp", alpha.con$Type),]
median(alpha.con.sub$Shannon)
alpha.con.sub <- alpha.con[grep("2.Ep", alpha.con$Type),]
median(alpha.con.sub$Shannon)

p <- ggplot(alpha.con, aes(Type, Shannon))
p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.con.dots",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.con.dots",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)

p + geom_boxplot(aes(fill = Type), outlier.colour="red", outlier.shape=NA, outlier.size=2) +
  ggtitle(plottitle) +
  coord_cartesian(ylim = c(0.3, 5.5)) +
  # geom_jitter(color="darkblue", shape=16, position=position_jitter(0.2)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.con",Sys.Date(),"png",sep = "."),width = 5, height = 5, dpi = 300)
ggsave(paste(PLOTS_DIR,"/alpha.boxplot.p.con",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 5, dpi = 300)
```

#### 5.2.2 beta diversity - PCoA
```{r}
bplot.obj <- capscale(otu_table(physeq4norm.ansys) ~ 1, distance = "bray")

bplot.scores <- scores(bplot.obj)
explained <- eigenvals( bplot.obj )/sum( eigenvals( bplot.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

bplot.df <- data.frame(id = row.names(bplot.scores$sites), bplot.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(bplot.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.single(bplot.df , components = c("PC1", "PC2"), groups = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.con.Type",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.con.Type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)

# Type & Ant
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Ant", treatment = "Type", variance = TRUE, connect = FALSE)

# Type & Type
p <- PlotOrd(bplot.df , components = c("PC1", "PC2"), groups = "Type", treatment = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.con.TypeType",Sys.Date(),"png",sep = "."),width = 7, height = 6, dpi = 300)
# ggsave(paste(PLOTS_DIR,"/plot.PCoA.p.con.TypeType",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 7, height = 6, dpi = 300)
```

#### 5.2.3 drivers

##### adonis
```{r}
(m.p.Type <- adonis(otu_table(physeq4norm.ansys) ~ Type,
                    data = get_variable(physeq4norm.ansys), method = "bray", permutations = 9999))
```

##### plotting - dbRDA
```{r}
dbRDA.obj <- capscale((otu_table(physeq4norm.ansys)) ~ Type, data=get_variable(physeq4norm.ansys),  distance = "bray")

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, Name = get_variable(physeq4norm.ansys)$Name, Type = get_variable(physeq4norm.ansys)$Type, Year = get_variable(physeq4norm.ansys)$Year, Ant = get_variable(physeq4norm.ansys)$Ant, Age = get_variable(physeq4norm.ansys)$Age, Tree = get_variable(physeq4norm.ansys)$TreeR)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

# Type
p <- PlotOrd.single(dbRDA.df , components = c("PC1", "PC2"), groups = "Type", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste(PLOTS_DIR,"/plot.dbRDA.p.con.Type",Sys.Date(),"png",sep = "."),width = 5, height = 4, dpi = 300)
ggsave(paste(PLOTS_DIR,"/plot.dbRDA.p.con.Type",Sys.Date(),"pdf",sep = "."), useDingbats=FALSE,width = 5, height = 4, dpi = 300)
```